diff --git a/src/test/java/org/apache/commons/lang3/builder/ReflectionDiffBuilderTest.java b/src/test/java/org/apache/commons/lang3/builder/ReflectionDiffBuilderTest.java
index 202bb21fa72..e2e134d9dcc 100644
--- a/src/test/java/org/apache/commons/lang3/builder/ReflectionDiffBuilderTest.java
+++ b/src/test/java/org/apache/commons/lang3/builder/ReflectionDiffBuilderTest.java
@@ -287,7 +287,7 @@ void testTransientFieldDifference() {
         final TypeTestClass firstObject = new TypeTestClass();
         firstObject.transientField = "a";
         final TypeTestClass secondObject = new TypeTestClass();
-        firstObject.transientField = "b";
+        secondObject.transientField = "b";
         DiffResult<TypeTestClass> list = firstObject.diff(secondObject);
         assertEquals(0, list.getNumberOfDiffs());
         list = firstObject.diffDeprecated(secondObject);
