diff --git a/src/main/java/org/apache/commons/lang3/EnumUtils.java b/src/main/java/org/apache/commons/lang3/EnumUtils.java
index 36cf44d385e..513b3140ae5 100644
--- a/src/main/java/org/apache/commons/lang3/EnumUtils.java
+++ b/src/main/java/org/apache/commons/lang3/EnumUtils.java
@@ -343,7 +343,7 @@ public static <E extends Enum<E>> E getEnumSystemProperty(final Class<E> enumCla
      * @since 3.18.0
      */
     public static <E extends Enum<E>> E getFirstEnum(final Class<E> enumClass, final int value, final ToIntFunction<E> toIntFunction, final E defaultEnum) {
-        if (isEnum(enumClass)) {
+        if (!isEnum(enumClass)) {
             return defaultEnum;
         }
         return stream(enumClass).filter(e -> value == toIntFunction.applyAsInt(e)).findFirst().orElse(defaultEnum);
@@ -372,7 +372,7 @@ public static <E extends Enum<E>> E getFirstEnumIgnoreCase(final Class<E> enumCl
     }
 
     private static <E extends Enum<E>> boolean isEnum(final Class<E> enumClass) {
-        return enumClass != null && !enumClass.isEnum();
+        return enumClass != null && enumClass.isEnum();
     }
 
     /**
