diff --git a/android/guava-testlib/src/com/google/common/collect/testing/google/SetGenerators.java b/android/guava-testlib/src/com/google/common/collect/testing/google/SetGenerators.java
index 481a2dadaece..3426a8ce2fc9 100644
--- a/android/guava-testlib/src/com/google/common/collect/testing/google/SetGenerators.java
+++ b/android/guava-testlib/src/com/google/common/collect/testing/google/SetGenerators.java
@@ -412,7 +412,7 @@ public List<Integer> order(List<Integer> insertionOrder) {
 
   private abstract static class AbstractContiguousSetGenerator
       extends TestIntegerSortedSetGenerator {
-    final ContiguousSet<Integer> checkedCreate(SortedSet<Integer> elementsSet) {
+    static final ContiguousSet<Integer> checkedCreate(SortedSet<Integer> elementsSet) {
       List<Integer> elements = new ArrayList<>(elementsSet);
       /*
        * A ContiguousSet can't have holes. If a test demands a hole, it should be changed so that it
diff --git a/android/guava-testlib/src/com/google/common/collect/testing/google/SortedMultisetTestSuiteBuilder.java b/android/guava-testlib/src/com/google/common/collect/testing/google/SortedMultisetTestSuiteBuilder.java
index 3fb03bb55dcd..f04fc4e17660 100644
--- a/android/guava-testlib/src/com/google/common/collect/testing/google/SortedMultisetTestSuiteBuilder.java
+++ b/android/guava-testlib/src/com/google/common/collect/testing/google/SortedMultisetTestSuiteBuilder.java
@@ -224,7 +224,7 @@ public SortedMultiset<E> create(Object... entries) {
    * work for this purpose, which may cause problems for navigable maps with non-string or unicode
    * generators.
    */
-  private List<String> getExtremeValues() {
+  private static List<String> getExtremeValues() {
     List<String> result = new ArrayList<>();
     result.add("!! a");
     result.add("!! b");
diff --git a/android/guava-testlib/src/com/google/common/collect/testing/testers/CollectionToArrayTester.java b/android/guava-testlib/src/com/google/common/collect/testing/testers/CollectionToArrayTester.java
index ef8bf08377ed..8d6c3472142d 100644
--- a/android/guava-testlib/src/com/google/common/collect/testing/testers/CollectionToArrayTester.java
+++ b/android/guava-testlib/src/com/google/common/collect/testing/testers/CollectionToArrayTester.java
@@ -186,7 +186,7 @@ public void testToArray_emptyArrayOfWrongTypeForEmptyCollection() {
         collection.toArray(array));
   }
 
-  private void expectArrayContentsAnyOrder(Object[] expected, Object[] actual) {
+  private static void expectArrayContentsAnyOrder(Object[] expected, Object[] actual) {
     assertEqualIgnoringOrder(asList(expected), asList(actual));
   }
 
diff --git a/android/guava/src/com/google/common/io/ByteSource.java b/android/guava/src/com/google/common/io/ByteSource.java
index d64960816ef9..e0695afe4e0f 100644
--- a/android/guava/src/com/google/common/io/ByteSource.java
+++ b/android/guava/src/com/google/common/io/ByteSource.java
@@ -229,7 +229,7 @@ public long size() throws IOException {
   }
 
   /** Counts the bytes in the given input stream using skip if possible. */
-  private long countBySkipping(InputStream in) throws IOException {
+  private static long countBySkipping(InputStream in) throws IOException {
     long count = 0;
     long skipped;
     while ((skipped = skipUpTo(in, Integer.MAX_VALUE)) > 0) {
diff --git a/android/guava/src/com/google/common/io/CharSource.java b/android/guava/src/com/google/common/io/CharSource.java
index 47957af13fe9..49d0d47cbdbf 100644
--- a/android/guava/src/com/google/common/io/CharSource.java
+++ b/android/guava/src/com/google/common/io/CharSource.java
@@ -229,7 +229,7 @@ public long length() throws IOException {
     }
   }
 
-  private long countBySkipping(Reader reader) throws IOException {
+  private static long countBySkipping(Reader reader) throws IOException {
     long count = 0;
     long read;
     while ((read = reader.skip(Long.MAX_VALUE)) != 0) {
diff --git a/android/guava/src/com/google/common/math/PairedStatsAccumulator.java b/android/guava/src/com/google/common/math/PairedStatsAccumulator.java
index fd691a1fd6d2..30cc0ccb81e9 100644
--- a/android/guava/src/com/google/common/math/PairedStatsAccumulator.java
+++ b/android/guava/src/com/google/common/math/PairedStatsAccumulator.java
@@ -230,7 +230,7 @@ public final LinearTransformation leastSquaresFit() {
     }
   }
 
-  private double ensurePositive(double value) {
+  private static double ensurePositive(double value) {
     if (value > 0.0) {
       return value;
     } else {
diff --git a/android/guava/src/com/google/common/util/concurrent/SimpleTimeLimiter.java b/android/guava/src/com/google/common/util/concurrent/SimpleTimeLimiter.java
index 622a0def3995..c898a1cc7649 100644
--- a/android/guava/src/com/google/common/util/concurrent/SimpleTimeLimiter.java
+++ b/android/guava/src/com/google/common/util/concurrent/SimpleTimeLimiter.java
@@ -258,7 +258,8 @@ private static boolean declaresInterruptedEx(Method method) {
     return false;
   }
 
-  private void wrapAndThrowExecutionExceptionOrError(Throwable cause) throws ExecutionException {
+  private static void wrapAndThrowExecutionExceptionOrError(Throwable cause)
+      throws ExecutionException {
     if (cause instanceof Error) {
       throw new ExecutionError((Error) cause);
     } else if (cause instanceof RuntimeException) {
@@ -268,7 +269,7 @@ private void wrapAndThrowExecutionExceptionOrError(Throwable cause) throws Execu
     }
   }
 
-  private void wrapAndThrowRuntimeExecutionExceptionOrError(Throwable cause) {
+  private static void wrapAndThrowRuntimeExecutionExceptionOrError(Throwable cause) {
     if (cause instanceof Error) {
       throw new ExecutionError((Error) cause);
     } else {
diff --git a/guava-testlib/src/com/google/common/collect/testing/TestsForMapsInJavaUtil.java b/guava-testlib/src/com/google/common/collect/testing/TestsForMapsInJavaUtil.java
index cbfb658408c1..d9d4a29547f4 100644
--- a/guava-testlib/src/com/google/common/collect/testing/TestsForMapsInJavaUtil.java
+++ b/guava-testlib/src/com/google/common/collect/testing/TestsForMapsInJavaUtil.java
@@ -101,11 +101,11 @@ protected Collection<Method> suppressForEmptyMap() {
     return emptySet();
   }
 
-  private Collection<Method> suppressForEmptyNavigableMap() {
+  private static Collection<Method> suppressForEmptyNavigableMap() {
     return emptySet();
   }
 
-  private Collection<Method> suppressForEmptySortedMap() {
+  private static Collection<Method> suppressForEmptySortedMap() {
     return emptySet();
   }
 
diff --git a/guava-testlib/src/com/google/common/collect/testing/google/SetGenerators.java b/guava-testlib/src/com/google/common/collect/testing/google/SetGenerators.java
index 481a2dadaece..3426a8ce2fc9 100644
--- a/guava-testlib/src/com/google/common/collect/testing/google/SetGenerators.java
+++ b/guava-testlib/src/com/google/common/collect/testing/google/SetGenerators.java
@@ -412,7 +412,7 @@ public List<Integer> order(List<Integer> insertionOrder) {
 
   private abstract static class AbstractContiguousSetGenerator
       extends TestIntegerSortedSetGenerator {
-    final ContiguousSet<Integer> checkedCreate(SortedSet<Integer> elementsSet) {
+    static final ContiguousSet<Integer> checkedCreate(SortedSet<Integer> elementsSet) {
       List<Integer> elements = new ArrayList<>(elementsSet);
       /*
        * A ContiguousSet can't have holes. If a test demands a hole, it should be changed so that it
diff --git a/guava-testlib/src/com/google/common/collect/testing/google/SortedMultisetTestSuiteBuilder.java b/guava-testlib/src/com/google/common/collect/testing/google/SortedMultisetTestSuiteBuilder.java
index 3fb03bb55dcd..f04fc4e17660 100644
--- a/guava-testlib/src/com/google/common/collect/testing/google/SortedMultisetTestSuiteBuilder.java
+++ b/guava-testlib/src/com/google/common/collect/testing/google/SortedMultisetTestSuiteBuilder.java
@@ -224,7 +224,7 @@ public SortedMultiset<E> create(Object... entries) {
    * work for this purpose, which may cause problems for navigable maps with non-string or unicode
    * generators.
    */
-  private List<String> getExtremeValues() {
+  private static List<String> getExtremeValues() {
     List<String> result = new ArrayList<>();
     result.add("!! a");
     result.add("!! b");
diff --git a/guava-testlib/src/com/google/common/collect/testing/testers/CollectionToArrayTester.java b/guava-testlib/src/com/google/common/collect/testing/testers/CollectionToArrayTester.java
index ef8bf08377ed..8d6c3472142d 100644
--- a/guava-testlib/src/com/google/common/collect/testing/testers/CollectionToArrayTester.java
+++ b/guava-testlib/src/com/google/common/collect/testing/testers/CollectionToArrayTester.java
@@ -186,7 +186,7 @@ public void testToArray_emptyArrayOfWrongTypeForEmptyCollection() {
         collection.toArray(array));
   }
 
-  private void expectArrayContentsAnyOrder(Object[] expected, Object[] actual) {
+  private static void expectArrayContentsAnyOrder(Object[] expected, Object[] actual) {
     assertEqualIgnoringOrder(asList(expected), asList(actual));
   }
 
diff --git a/guava/src/com/google/common/io/ByteSource.java b/guava/src/com/google/common/io/ByteSource.java
index d64960816ef9..e0695afe4e0f 100644
--- a/guava/src/com/google/common/io/ByteSource.java
+++ b/guava/src/com/google/common/io/ByteSource.java
@@ -229,7 +229,7 @@ public long size() throws IOException {
   }
 
   /** Counts the bytes in the given input stream using skip if possible. */
-  private long countBySkipping(InputStream in) throws IOException {
+  private static long countBySkipping(InputStream in) throws IOException {
     long count = 0;
     long skipped;
     while ((skipped = skipUpTo(in, Integer.MAX_VALUE)) > 0) {
diff --git a/guava/src/com/google/common/io/CharSource.java b/guava/src/com/google/common/io/CharSource.java
index b397d192cf4f..e6e224462a50 100644
--- a/guava/src/com/google/common/io/CharSource.java
+++ b/guava/src/com/google/common/io/CharSource.java
@@ -227,7 +227,7 @@ public long length() throws IOException {
     }
   }
 
-  private long countBySkipping(Reader reader) throws IOException {
+  private static long countBySkipping(Reader reader) throws IOException {
     long count = 0;
     long read;
     while ((read = reader.skip(Long.MAX_VALUE)) != 0) {
diff --git a/guava/src/com/google/common/math/PairedStatsAccumulator.java b/guava/src/com/google/common/math/PairedStatsAccumulator.java
index fd691a1fd6d2..30cc0ccb81e9 100644
--- a/guava/src/com/google/common/math/PairedStatsAccumulator.java
+++ b/guava/src/com/google/common/math/PairedStatsAccumulator.java
@@ -230,7 +230,7 @@ public final LinearTransformation leastSquaresFit() {
     }
   }
 
-  private double ensurePositive(double value) {
+  private static double ensurePositive(double value) {
     if (value > 0.0) {
       return value;
     } else {
diff --git a/guava/src/com/google/common/util/concurrent/SimpleTimeLimiter.java b/guava/src/com/google/common/util/concurrent/SimpleTimeLimiter.java
index 622a0def3995..c898a1cc7649 100644
--- a/guava/src/com/google/common/util/concurrent/SimpleTimeLimiter.java
+++ b/guava/src/com/google/common/util/concurrent/SimpleTimeLimiter.java
@@ -258,7 +258,8 @@ private static boolean declaresInterruptedEx(Method method) {
     return false;
   }
 
-  private void wrapAndThrowExecutionExceptionOrError(Throwable cause) throws ExecutionException {
+  private static void wrapAndThrowExecutionExceptionOrError(Throwable cause)
+      throws ExecutionException {
     if (cause instanceof Error) {
       throw new ExecutionError((Error) cause);
     } else if (cause instanceof RuntimeException) {
@@ -268,7 +269,7 @@ private void wrapAndThrowExecutionExceptionOrError(Throwable cause) throws Execu
     }
   }
 
-  private void wrapAndThrowRuntimeExecutionExceptionOrError(Throwable cause) {
+  private static void wrapAndThrowRuntimeExecutionExceptionOrError(Throwable cause) {
     if (cause instanceof Error) {
       throw new ExecutionError((Error) cause);
     } else {
