diff --git a/android/guava/src/com/google/common/base/NullnessCasts.java b/android/guava/src/com/google/common/base/NullnessCasts.java
index 5c99948ad47c..e46e8e7ac6fa 100644
--- a/android/guava/src/com/google/common/base/NullnessCasts.java
+++ b/android/guava/src/com/google/common/base/NullnessCasts.java
@@ -44,9 +44,11 @@ final class NullnessCasts {
    * doesn't work: Because nullness analyses typically infer the nullness of local variables,
    * there's no way to assign a {@code @Nullable T} to a field {@code T foo;} and instruct the
    * analysis that that means "plain {@code T}" rather than the inferred type {@code @Nullable T}.
-   * (Even if supported added {@code @NonNull}, that would not help, since the problem case
-   * addressed by this method is the case in which {@code T} has parametric nullness -- and thus its
-   * value may be legitimately {@code null}.)
+   * (And even if annotations on local variables were permitted as an optional hint, no annotation
+   * would be the right tool for the job here: {@code @Nullable} is the annotation that we're trying
+   * to get rid of, and {@code @NonNull} would be wrong for our use case for the same reason as
+   * {@code requireNonNull}: Our use case is the one in which {@code T} has parametric nullness—and
+   * thus its value may be legitimately {@code null}.)
    */
   @ParametricNullness
   @SuppressWarnings("nullness")
diff --git a/android/guava/src/com/google/common/collect/NullnessCasts.java b/android/guava/src/com/google/common/collect/NullnessCasts.java
index 82bcc2dbeaef..6863dbfbfd7a 100644
--- a/android/guava/src/com/google/common/collect/NullnessCasts.java
+++ b/android/guava/src/com/google/common/collect/NullnessCasts.java
@@ -44,9 +44,11 @@ final class NullnessCasts {
    * doesn't work: Because nullness analyses typically infer the nullness of local variables,
    * there's no way to assign a {@code @Nullable T} to a field {@code T foo;} and instruct the
    * analysis that that means "plain {@code T}" rather than the inferred type {@code @Nullable T}.
-   * (Even if supported added {@code @NonNull}, that would not help, since the problem case
-   * addressed by this method is the case in which {@code T} has parametric nullness -- and thus its
-   * value may be legitimately {@code null}.)
+   * (And even if annotations on local variables were permitted as an optional hint, no annotation
+   * would be the right tool for the job here: {@code @Nullable} is the annotation that we're trying
+   * to get rid of, and {@code @NonNull} would be wrong for our use case for the same reason as
+   * {@code requireNonNull}: Our use case is the one in which {@code T} has parametric nullness—and
+   * thus its value may be legitimately {@code null}.)
    */
   @ParametricNullness
   @SuppressWarnings("nullness")
diff --git a/android/guava/src/com/google/common/util/concurrent/NullnessCasts.java b/android/guava/src/com/google/common/util/concurrent/NullnessCasts.java
index 7ef866d26f1e..f20d67dd681f 100644
--- a/android/guava/src/com/google/common/util/concurrent/NullnessCasts.java
+++ b/android/guava/src/com/google/common/util/concurrent/NullnessCasts.java
@@ -44,9 +44,11 @@ final class NullnessCasts {
    * doesn't work: Because nullness analyses typically infer the nullness of local variables,
    * there's no way to assign a {@code @Nullable T} to a field {@code T foo;} and instruct the
    * analysis that that means "plain {@code T}" rather than the inferred type {@code @Nullable T}.
-   * (Even if supported added {@code @NonNull}, that would not help, since the problem case
-   * addressed by this method is the case in which {@code T} has parametric nullness -- and thus its
-   * value may be legitimately {@code null}.)
+   * (And even if annotations on local variables were permitted as an optional hint, no annotation
+   * would be the right tool for the job here: {@code @Nullable} is the annotation that we're trying
+   * to get rid of, and {@code @NonNull} would be wrong for our use case for the same reason as
+   * {@code requireNonNull}: Our use case is the one in which {@code T} has parametric nullness—and
+   * thus its value may be legitimately {@code null}.)
    */
   @SuppressWarnings("nullness")
   @ParametricNullness
diff --git a/guava/src/com/google/common/base/NullnessCasts.java b/guava/src/com/google/common/base/NullnessCasts.java
index 5c99948ad47c..e46e8e7ac6fa 100644
--- a/guava/src/com/google/common/base/NullnessCasts.java
+++ b/guava/src/com/google/common/base/NullnessCasts.java
@@ -44,9 +44,11 @@ final class NullnessCasts {
    * doesn't work: Because nullness analyses typically infer the nullness of local variables,
    * there's no way to assign a {@code @Nullable T} to a field {@code T foo;} and instruct the
    * analysis that that means "plain {@code T}" rather than the inferred type {@code @Nullable T}.
-   * (Even if supported added {@code @NonNull}, that would not help, since the problem case
-   * addressed by this method is the case in which {@code T} has parametric nullness -- and thus its
-   * value may be legitimately {@code null}.)
+   * (And even if annotations on local variables were permitted as an optional hint, no annotation
+   * would be the right tool for the job here: {@code @Nullable} is the annotation that we're trying
+   * to get rid of, and {@code @NonNull} would be wrong for our use case for the same reason as
+   * {@code requireNonNull}: Our use case is the one in which {@code T} has parametric nullness—and
+   * thus its value may be legitimately {@code null}.)
    */
   @ParametricNullness
   @SuppressWarnings("nullness")
diff --git a/guava/src/com/google/common/collect/NullnessCasts.java b/guava/src/com/google/common/collect/NullnessCasts.java
index 82bcc2dbeaef..6863dbfbfd7a 100644
--- a/guava/src/com/google/common/collect/NullnessCasts.java
+++ b/guava/src/com/google/common/collect/NullnessCasts.java
@@ -44,9 +44,11 @@ final class NullnessCasts {
    * doesn't work: Because nullness analyses typically infer the nullness of local variables,
    * there's no way to assign a {@code @Nullable T} to a field {@code T foo;} and instruct the
    * analysis that that means "plain {@code T}" rather than the inferred type {@code @Nullable T}.
-   * (Even if supported added {@code @NonNull}, that would not help, since the problem case
-   * addressed by this method is the case in which {@code T} has parametric nullness -- and thus its
-   * value may be legitimately {@code null}.)
+   * (And even if annotations on local variables were permitted as an optional hint, no annotation
+   * would be the right tool for the job here: {@code @Nullable} is the annotation that we're trying
+   * to get rid of, and {@code @NonNull} would be wrong for our use case for the same reason as
+   * {@code requireNonNull}: Our use case is the one in which {@code T} has parametric nullness—and
+   * thus its value may be legitimately {@code null}.)
    */
   @ParametricNullness
   @SuppressWarnings("nullness")
diff --git a/guava/src/com/google/common/util/concurrent/NullnessCasts.java b/guava/src/com/google/common/util/concurrent/NullnessCasts.java
index 7ef866d26f1e..f20d67dd681f 100644
--- a/guava/src/com/google/common/util/concurrent/NullnessCasts.java
+++ b/guava/src/com/google/common/util/concurrent/NullnessCasts.java
@@ -44,9 +44,11 @@ final class NullnessCasts {
    * doesn't work: Because nullness analyses typically infer the nullness of local variables,
    * there's no way to assign a {@code @Nullable T} to a field {@code T foo;} and instruct the
    * analysis that that means "plain {@code T}" rather than the inferred type {@code @Nullable T}.
-   * (Even if supported added {@code @NonNull}, that would not help, since the problem case
-   * addressed by this method is the case in which {@code T} has parametric nullness -- and thus its
-   * value may be legitimately {@code null}.)
+   * (And even if annotations on local variables were permitted as an optional hint, no annotation
+   * would be the right tool for the job here: {@code @Nullable} is the annotation that we're trying
+   * to get rid of, and {@code @NonNull} would be wrong for our use case for the same reason as
+   * {@code requireNonNull}: Our use case is the one in which {@code T} has parametric nullness—and
+   * thus its value may be legitimately {@code null}.)
    */
   @SuppressWarnings("nullness")
   @ParametricNullness
