diff --git a/android/guava/src/com/google/common/eventbus/SubscriberExceptionContext.java b/android/guava/src/com/google/common/eventbus/SubscriberExceptionContext.java
index f6beaad511d7..df7b66d6deed 100644
--- a/android/guava/src/com/google/common/eventbus/SubscriberExceptionContext.java
+++ b/android/guava/src/com/google/common/eventbus/SubscriberExceptionContext.java
@@ -45,30 +45,24 @@ public class SubscriberExceptionContext {
   }
 
   /**
-   * @return The {@link EventBus} that handled the event and the subscriber. Useful for broadcasting
-   *     a new event based on the error.
+   * Returns the {@link EventBus} that handled the event and the subscriber. Useful for broadcasting
+   * a new event based on the error.
    */
   public EventBus getEventBus() {
     return eventBus;
   }
 
-  /**
-   * @return The event object that caused the subscriber to throw.
-   */
+  /** Returns the event object that caused the subscriber to throw. */
   public Object getEvent() {
     return event;
   }
 
-  /**
-   * @return The object context that the subscriber was called on.
-   */
+  /** Returns the object context that the subscriber was called on. */
   public Object getSubscriber() {
     return subscriber;
   }
 
-  /**
-   * @return The subscribed method that threw the exception.
-   */
+  /** Returns the subscribed method that threw the exception. */
   public Method getSubscriberMethod() {
     return subscriberMethod;
   }
diff --git a/android/guava/src/com/google/common/math/IntMath.java b/android/guava/src/com/google/common/math/IntMath.java
index 8a19ffa75938..b2957cdcd0b8 100644
--- a/android/guava/src/com/google/common/math/IntMath.java
+++ b/android/guava/src/com/google/common/math/IntMath.java
@@ -517,10 +517,10 @@ public static int checkedPow(int b, int k) {
         case 0:
           return accum;
         case 1:
-          return checkedMultiply(accum, b);
+          return Math.multiplyExact(accum, b);
         default:
           if ((k & 1) != 0) {
-            accum = checkedMultiply(accum, b);
+            accum = Math.multiplyExact(accum, b);
           }
           k >>= 1;
           if (k > 0) {
diff --git a/android/guava/src/com/google/common/math/LongMath.java b/android/guava/src/com/google/common/math/LongMath.java
index 272791a649bc..c1c91cdf9732 100644
--- a/android/guava/src/com/google/common/math/LongMath.java
+++ b/android/guava/src/com/google/common/math/LongMath.java
@@ -615,10 +615,10 @@ public static long checkedPow(long b, int k) {
         case 0:
           return accum;
         case 1:
-          return checkedMultiply(accum, b);
+          return Math.multiplyExact(accum, b);
         default:
           if ((k & 1) != 0) {
-            accum = checkedMultiply(accum, b);
+            accum = Math.multiplyExact(accum, b);
           }
           k >>= 1;
           if (k > 0) {
diff --git a/android/guava/src/com/google/common/util/concurrent/CycleDetectingLockFactory.java b/android/guava/src/com/google/common/util/concurrent/CycleDetectingLockFactory.java
index 5040b64eb192..a59f70959c52 100644
--- a/android/guava/src/com/google/common/util/concurrent/CycleDetectingLockFactory.java
+++ b/android/guava/src/com/google/common/util/concurrent/CycleDetectingLockFactory.java
@@ -15,6 +15,7 @@
 package com.google.common.util.concurrent;
 
 import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.collect.Lists.newArrayListWithCapacity;
 import static java.util.Objects.requireNonNull;
 
 import com.google.common.annotations.GwtIncompatible;
@@ -459,11 +460,11 @@ private CycleDetectingLockFactory(Policy policy) {
    */
   // This is logically a Set, but an ArrayList is used to minimize the amount
   // of allocation done on lock()/unlock().
-  private static final ThreadLocal<ArrayList<LockGraphNode>> acquiredLocks =
-      new ThreadLocal<ArrayList<LockGraphNode>>() {
+  private static final ThreadLocal<List<LockGraphNode>> acquiredLocks =
+      new ThreadLocal<List<LockGraphNode>>() {
         @Override
-        protected ArrayList<LockGraphNode> initialValue() {
-          return Lists.<LockGraphNode>newArrayListWithCapacity(3);
+        protected List<LockGraphNode> initialValue() {
+          return newArrayListWithCapacity(3);
         }
       };
 
@@ -713,7 +714,7 @@ void checkAcquiredLock(Policy policy, LockGraphNode acquiredLock) {
   private void aboutToAcquire(CycleDetectingLock lock) {
     if (!lock.isAcquiredByCurrentThread()) {
       // requireNonNull accommodates Android's @RecentlyNullable annotation on ThreadLocal.get
-      ArrayList<LockGraphNode> acquiredLockList = requireNonNull(acquiredLocks.get());
+      List<LockGraphNode> acquiredLockList = requireNonNull(acquiredLocks.get());
       LockGraphNode node = lock.getLockGraphNode();
       node.checkAcquiredLocks(policy, acquiredLockList);
       acquiredLockList.add(node);
@@ -728,7 +729,7 @@ private void aboutToAcquire(CycleDetectingLock lock) {
   private static void lockStateChanged(CycleDetectingLock lock) {
     if (!lock.isAcquiredByCurrentThread()) {
       // requireNonNull accommodates Android's @RecentlyNullable annotation on ThreadLocal.get
-      ArrayList<LockGraphNode> acquiredLockList = requireNonNull(acquiredLocks.get());
+      List<LockGraphNode> acquiredLockList = requireNonNull(acquiredLocks.get());
       LockGraphNode node = lock.getLockGraphNode();
       // Iterate in reverse because locks are usually locked/unlocked in a
       // LIFO order.
diff --git a/android/guava/src/com/google/common/xml/XmlEscapers.java b/android/guava/src/com/google/common/xml/XmlEscapers.java
index 9ea51415b251..72782eee1e61 100644
--- a/android/guava/src/com/google/common/xml/XmlEscapers.java
+++ b/android/guava/src/com/google/common/xml/XmlEscapers.java
@@ -97,8 +97,6 @@ public static Escaper xmlContentEscaper() {
   public static Escaper xmlAttributeEscaper() {
     return XML_ATTRIBUTE_ESCAPER;
   }
-
-  private static final Escaper XML_ESCAPER;
   private static final Escaper XML_CONTENT_ESCAPER;
   private static final Escaper XML_ATTRIBUTE_ESCAPER;
 
@@ -134,7 +132,6 @@ public static Escaper xmlAttributeEscaper() {
     XML_CONTENT_ESCAPER = builder.build();
     builder.addEscape('\'', "&apos;");
     builder.addEscape('"', "&quot;");
-    XML_ESCAPER = builder.build();
     builder.addEscape('\t', "&#x9;");
     builder.addEscape('\n', "&#xA;");
     builder.addEscape('\r', "&#xD;");
diff --git a/guava/src/com/google/common/base/Platform.java b/guava/src/com/google/common/base/Platform.java
index 5be90156fe7b..6b1728db23c5 100644
--- a/guava/src/com/google/common/base/Platform.java
+++ b/guava/src/com/google/common/base/Platform.java
@@ -18,7 +18,6 @@
 import com.google.common.annotations.GwtCompatible;
 import java.lang.ref.WeakReference;
 import java.util.Locale;
-import java.util.logging.Logger;
 import java.util.regex.Pattern;
 import org.jspecify.annotations.Nullable;
 
@@ -29,7 +28,6 @@
  */
 @GwtCompatible
 final class Platform {
-  private static final Logger logger = Logger.getLogger(Platform.class.getName());
   private static final PatternCompiler patternCompiler = loadPatternCompiler();
 
   private Platform() {}
diff --git a/guava/src/com/google/common/eventbus/SubscriberExceptionContext.java b/guava/src/com/google/common/eventbus/SubscriberExceptionContext.java
index f6beaad511d7..df7b66d6deed 100644
--- a/guava/src/com/google/common/eventbus/SubscriberExceptionContext.java
+++ b/guava/src/com/google/common/eventbus/SubscriberExceptionContext.java
@@ -45,30 +45,24 @@ public class SubscriberExceptionContext {
   }
 
   /**
-   * @return The {@link EventBus} that handled the event and the subscriber. Useful for broadcasting
-   *     a new event based on the error.
+   * Returns the {@link EventBus} that handled the event and the subscriber. Useful for broadcasting
+   * a new event based on the error.
    */
   public EventBus getEventBus() {
     return eventBus;
   }
 
-  /**
-   * @return The event object that caused the subscriber to throw.
-   */
+  /** Returns the event object that caused the subscriber to throw. */
   public Object getEvent() {
     return event;
   }
 
-  /**
-   * @return The object context that the subscriber was called on.
-   */
+  /** Returns the object context that the subscriber was called on. */
   public Object getSubscriber() {
     return subscriber;
   }
 
-  /**
-   * @return The subscribed method that threw the exception.
-   */
+  /** Returns the subscribed method that threw the exception. */
   public Method getSubscriberMethod() {
     return subscriberMethod;
   }
diff --git a/guava/src/com/google/common/math/IntMath.java b/guava/src/com/google/common/math/IntMath.java
index 8a19ffa75938..b2957cdcd0b8 100644
--- a/guava/src/com/google/common/math/IntMath.java
+++ b/guava/src/com/google/common/math/IntMath.java
@@ -517,10 +517,10 @@ public static int checkedPow(int b, int k) {
         case 0:
           return accum;
         case 1:
-          return checkedMultiply(accum, b);
+          return Math.multiplyExact(accum, b);
         default:
           if ((k & 1) != 0) {
-            accum = checkedMultiply(accum, b);
+            accum = Math.multiplyExact(accum, b);
           }
           k >>= 1;
           if (k > 0) {
diff --git a/guava/src/com/google/common/math/LongMath.java b/guava/src/com/google/common/math/LongMath.java
index 272791a649bc..c1c91cdf9732 100644
--- a/guava/src/com/google/common/math/LongMath.java
+++ b/guava/src/com/google/common/math/LongMath.java
@@ -615,10 +615,10 @@ public static long checkedPow(long b, int k) {
         case 0:
           return accum;
         case 1:
-          return checkedMultiply(accum, b);
+          return Math.multiplyExact(accum, b);
         default:
           if ((k & 1) != 0) {
-            accum = checkedMultiply(accum, b);
+            accum = Math.multiplyExact(accum, b);
           }
           k >>= 1;
           if (k > 0) {
diff --git a/guava/src/com/google/common/util/concurrent/CycleDetectingLockFactory.java b/guava/src/com/google/common/util/concurrent/CycleDetectingLockFactory.java
index 5040b64eb192..a59f70959c52 100644
--- a/guava/src/com/google/common/util/concurrent/CycleDetectingLockFactory.java
+++ b/guava/src/com/google/common/util/concurrent/CycleDetectingLockFactory.java
@@ -15,6 +15,7 @@
 package com.google.common.util.concurrent;
 
 import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.collect.Lists.newArrayListWithCapacity;
 import static java.util.Objects.requireNonNull;
 
 import com.google.common.annotations.GwtIncompatible;
@@ -459,11 +460,11 @@ private CycleDetectingLockFactory(Policy policy) {
    */
   // This is logically a Set, but an ArrayList is used to minimize the amount
   // of allocation done on lock()/unlock().
-  private static final ThreadLocal<ArrayList<LockGraphNode>> acquiredLocks =
-      new ThreadLocal<ArrayList<LockGraphNode>>() {
+  private static final ThreadLocal<List<LockGraphNode>> acquiredLocks =
+      new ThreadLocal<List<LockGraphNode>>() {
         @Override
-        protected ArrayList<LockGraphNode> initialValue() {
-          return Lists.<LockGraphNode>newArrayListWithCapacity(3);
+        protected List<LockGraphNode> initialValue() {
+          return newArrayListWithCapacity(3);
         }
       };
 
@@ -713,7 +714,7 @@ void checkAcquiredLock(Policy policy, LockGraphNode acquiredLock) {
   private void aboutToAcquire(CycleDetectingLock lock) {
     if (!lock.isAcquiredByCurrentThread()) {
       // requireNonNull accommodates Android's @RecentlyNullable annotation on ThreadLocal.get
-      ArrayList<LockGraphNode> acquiredLockList = requireNonNull(acquiredLocks.get());
+      List<LockGraphNode> acquiredLockList = requireNonNull(acquiredLocks.get());
       LockGraphNode node = lock.getLockGraphNode();
       node.checkAcquiredLocks(policy, acquiredLockList);
       acquiredLockList.add(node);
@@ -728,7 +729,7 @@ private void aboutToAcquire(CycleDetectingLock lock) {
   private static void lockStateChanged(CycleDetectingLock lock) {
     if (!lock.isAcquiredByCurrentThread()) {
       // requireNonNull accommodates Android's @RecentlyNullable annotation on ThreadLocal.get
-      ArrayList<LockGraphNode> acquiredLockList = requireNonNull(acquiredLocks.get());
+      List<LockGraphNode> acquiredLockList = requireNonNull(acquiredLocks.get());
       LockGraphNode node = lock.getLockGraphNode();
       // Iterate in reverse because locks are usually locked/unlocked in a
       // LIFO order.
diff --git a/guava/src/com/google/common/xml/XmlEscapers.java b/guava/src/com/google/common/xml/XmlEscapers.java
index 9ea51415b251..72782eee1e61 100644
--- a/guava/src/com/google/common/xml/XmlEscapers.java
+++ b/guava/src/com/google/common/xml/XmlEscapers.java
@@ -97,8 +97,6 @@ public static Escaper xmlContentEscaper() {
   public static Escaper xmlAttributeEscaper() {
     return XML_ATTRIBUTE_ESCAPER;
   }
-
-  private static final Escaper XML_ESCAPER;
   private static final Escaper XML_CONTENT_ESCAPER;
   private static final Escaper XML_ATTRIBUTE_ESCAPER;
 
@@ -134,7 +132,6 @@ public static Escaper xmlAttributeEscaper() {
     XML_CONTENT_ESCAPER = builder.build();
     builder.addEscape('\'', "&apos;");
     builder.addEscape('"', "&quot;");
-    XML_ESCAPER = builder.build();
     builder.addEscape('\t', "&#x9;");
     builder.addEscape('\n', "&#xA;");
     builder.addEscape('\r', "&#xD;");
