diff --git a/android/guava-tests/test/com/google/common/base/StringsTest.java b/android/guava-tests/test/com/google/common/base/StringsTest.java
index c5fd2f98bdb7..61de698e2e9f 100644
--- a/android/guava-tests/test/com/google/common/base/StringsTest.java
+++ b/android/guava-tests/test/com/google/common/base/StringsTest.java
@@ -213,6 +213,8 @@ public void testLenientFormat() {
     assertEquals("5 + 6 = 11", Strings.lenientFormat("5 + %s = 11", 6));
     assertEquals("5 + 6 = 11", Strings.lenientFormat("5 + 6 = %s", 11));
     assertEquals("5 + 6 = 11", Strings.lenientFormat("%s + %s = %s", 5, 6, 11));
+    assertEquals(
+        "5 + 6 = 11", Strings.lenientFormat("%s + %s = %s", (Object[]) new Integer[] {5, 6, 11}));
     assertEquals("null [null, null]", Strings.lenientFormat("%s", null, null, null));
     assertEquals("null [5, 6]", Strings.lenientFormat(null, 5, 6));
     assertEquals("null", Strings.lenientFormat("%s", (Object) null));
diff --git a/android/guava/src/com/google/common/base/Strings.java b/android/guava/src/com/google/common/base/Strings.java
index 0f26ec985fbb..feb5915ca2f8 100644
--- a/android/guava/src/com/google/common/base/Strings.java
+++ b/android/guava/src/com/google/common/base/Strings.java
@@ -263,10 +263,6 @@ public static String lenientFormat(
 
     if (args == null) {
       args = new Object[] {"(Object[])null"};
-    } else {
-      for (int i = 0; i < args.length; i++) {
-        args[i] = lenientToString(args[i]);
-      }
     }
 
     // start substituting the arguments into the '%s' placeholders
@@ -279,18 +275,18 @@ public static String lenientFormat(
         break;
       }
       builder.append(template, templateStart, placeholderStart);
-      builder.append(args[i++]);
+      builder.append(lenientToString(args[i++]));
       templateStart = placeholderStart + 2;
     }
     builder.append(template, templateStart, template.length());
 
-    // if we run out of placeholders, append the extra args in square braces
+    // if we run out of placeholders, append the extra args in square brackets
     if (i < args.length) {
-      builder.append(" [");
-      builder.append(args[i++]);
-      while (i < args.length) {
-        builder.append(", ");
-        builder.append(args[i++]);
+      String prefix = " [";
+      for (; i < args.length; i++) {
+        builder.append(prefix);
+        builder.append(lenientToString(args[i]));
+        prefix = ", ";
       }
       builder.append(']');
     }
diff --git a/guava-tests/test/com/google/common/base/StringsTest.java b/guava-tests/test/com/google/common/base/StringsTest.java
index c5fd2f98bdb7..61de698e2e9f 100644
--- a/guava-tests/test/com/google/common/base/StringsTest.java
+++ b/guava-tests/test/com/google/common/base/StringsTest.java
@@ -213,6 +213,8 @@ public void testLenientFormat() {
     assertEquals("5 + 6 = 11", Strings.lenientFormat("5 + %s = 11", 6));
     assertEquals("5 + 6 = 11", Strings.lenientFormat("5 + 6 = %s", 11));
     assertEquals("5 + 6 = 11", Strings.lenientFormat("%s + %s = %s", 5, 6, 11));
+    assertEquals(
+        "5 + 6 = 11", Strings.lenientFormat("%s + %s = %s", (Object[]) new Integer[] {5, 6, 11}));
     assertEquals("null [null, null]", Strings.lenientFormat("%s", null, null, null));
     assertEquals("null [5, 6]", Strings.lenientFormat(null, 5, 6));
     assertEquals("null", Strings.lenientFormat("%s", (Object) null));
diff --git a/guava/src/com/google/common/base/Strings.java b/guava/src/com/google/common/base/Strings.java
index 27c1076e374e..ff95a5268fd6 100644
--- a/guava/src/com/google/common/base/Strings.java
+++ b/guava/src/com/google/common/base/Strings.java
@@ -269,10 +269,6 @@ public static String lenientFormat(
 
     if (args == null) {
       args = new Object[] {"(Object[])null"};
-    } else {
-      for (int i = 0; i < args.length; i++) {
-        args[i] = lenientToString(args[i]);
-      }
     }
 
     // start substituting the arguments into the '%s' placeholders
@@ -285,18 +281,18 @@ public static String lenientFormat(
         break;
       }
       builder.append(template, templateStart, placeholderStart);
-      builder.append(args[i++]);
+      builder.append(lenientToString(args[i++]));
       templateStart = placeholderStart + 2;
     }
     builder.append(template, templateStart, template.length());
 
-    // if we run out of placeholders, append the extra args in square braces
+    // if we run out of placeholders, append the extra args in square brackets
     if (i < args.length) {
-      builder.append(" [");
-      builder.append(args[i++]);
-      while (i < args.length) {
-        builder.append(", ");
-        builder.append(args[i++]);
+      String prefix = " [";
+      for (; i < args.length; i++) {
+        builder.append(prefix);
+        builder.append(lenientToString(args[i]));
+        prefix = ", ";
       }
       builder.append(']');
     }
