diff --git a/guava-tests/test/com/google/common/cache/LocalCacheMapComputeTest.java b/guava-tests/test/com/google/common/cache/LocalCacheMapComputeTest.java
index a914e29b5eb7..54d8ae55ade5 100644
--- a/guava-tests/test/com/google/common/cache/LocalCacheMapComputeTest.java
+++ b/guava-tests/test/com/google/common/cache/LocalCacheMapComputeTest.java
@@ -125,6 +125,23 @@ public void onRemoval(RemovalNotification<Integer, Integer> notification) {
     CacheTesting.checkEmpty(cache);
   }
 
+  public void testComputeIfPresent_error() {
+    Cache<String, String> cache = CacheBuilder.newBuilder().build();
+    cache.put(key, "1");
+    assertThrows(
+        Error.class,
+        () ->
+            cache
+                .asMap()
+                .computeIfPresent(
+                    key,
+                    (k, v) -> {
+                      throw new Error();
+                    }));
+    assertThat(cache.getIfPresent(key)).isEqualTo("1");
+    assertThat(cache.asMap().computeIfPresent(key, (k, v) -> "2")).isEqualTo(null);
+  }
+
   public void testUpdates() {
     cache.put(key, "1");
     // simultaneous update for same key, some null, some non-null
