diff --git a/android/guava/src/com/google/common/primitives/Doubles.java b/android/guava/src/com/google/common/primitives/Doubles.java
index ce43ea648910..ec1cbe019b7f 100644
--- a/android/guava/src/com/google/common/primitives/Doubles.java
+++ b/android/guava/src/com/google/common/primitives/Doubles.java
@@ -58,8 +58,6 @@ private Doubles() {}
    *
    * @since 10.0
    */
-  // The constants value gets inlined here.
-  @SuppressWarnings("AndroidJdkLibsChecker")
   public static final int BYTES = Double.BYTES;
 
   /**
diff --git a/android/guava/src/com/google/common/primitives/Floats.java b/android/guava/src/com/google/common/primitives/Floats.java
index de0b7abda209..5266e20d9ac6 100644
--- a/android/guava/src/com/google/common/primitives/Floats.java
+++ b/android/guava/src/com/google/common/primitives/Floats.java
@@ -56,8 +56,6 @@ private Floats() {}
    *
    * @since 10.0
    */
-  // The constants value gets inlined here.
-  @SuppressWarnings("AndroidJdkLibsChecker")
   public static final int BYTES = Float.BYTES;
 
   /**
diff --git a/android/guava/src/com/google/common/primitives/Ints.java b/android/guava/src/com/google/common/primitives/Ints.java
index 9fd547e29197..f474a72d9ed5 100644
--- a/android/guava/src/com/google/common/primitives/Ints.java
+++ b/android/guava/src/com/google/common/primitives/Ints.java
@@ -56,8 +56,6 @@ private Ints() {}
    *
    * <p>Prefer {@link Integer#BYTES} instead.
    */
-  // The constants value gets inlined here.
-  @SuppressWarnings("AndroidJdkLibsChecker")
   public static final int BYTES = Integer.BYTES;
 
   /**
diff --git a/android/guava/src/com/google/common/primitives/Longs.java b/android/guava/src/com/google/common/primitives/Longs.java
index 49aa2ce338f5..9cc9972decf5 100644
--- a/android/guava/src/com/google/common/primitives/Longs.java
+++ b/android/guava/src/com/google/common/primitives/Longs.java
@@ -55,8 +55,6 @@ private Longs() {}
    *
    * <p>Prefer {@link Long#BYTES} instead.
    */
-  // The constants value gets inlined here.
-  @SuppressWarnings("AndroidJdkLibsChecker")
   public static final int BYTES = Long.BYTES;
 
   /**
diff --git a/android/guava/src/com/google/common/primitives/Shorts.java b/android/guava/src/com/google/common/primitives/Shorts.java
index 3a54130d8169..1e95a081b701 100644
--- a/android/guava/src/com/google/common/primitives/Shorts.java
+++ b/android/guava/src/com/google/common/primitives/Shorts.java
@@ -54,8 +54,6 @@ private Shorts() {}
    *
    * <p>Prefer {@link Short#BYTES} instead.
    */
-  // The constants value gets inlined here.
-  @SuppressWarnings("AndroidJdkLibsChecker")
   public static final int BYTES = Short.BYTES;
 
   /**
diff --git a/guava/src/com/google/common/primitives/Doubles.java b/guava/src/com/google/common/primitives/Doubles.java
index 821832b96f31..1fc33f9b1d68 100644
--- a/guava/src/com/google/common/primitives/Doubles.java
+++ b/guava/src/com/google/common/primitives/Doubles.java
@@ -58,8 +58,6 @@ private Doubles() {}
    *
    * @since 10.0
    */
-  // The constants value gets inlined here.
-  @SuppressWarnings("AndroidJdkLibsChecker")
   public static final int BYTES = Double.BYTES;
 
   /**
diff --git a/guava/src/com/google/common/primitives/Floats.java b/guava/src/com/google/common/primitives/Floats.java
index de0b7abda209..5266e20d9ac6 100644
--- a/guava/src/com/google/common/primitives/Floats.java
+++ b/guava/src/com/google/common/primitives/Floats.java
@@ -56,8 +56,6 @@ private Floats() {}
    *
    * @since 10.0
    */
-  // The constants value gets inlined here.
-  @SuppressWarnings("AndroidJdkLibsChecker")
   public static final int BYTES = Float.BYTES;
 
   /**
diff --git a/guava/src/com/google/common/primitives/Ints.java b/guava/src/com/google/common/primitives/Ints.java
index 22da95e63389..11611a804ab7 100644
--- a/guava/src/com/google/common/primitives/Ints.java
+++ b/guava/src/com/google/common/primitives/Ints.java
@@ -56,8 +56,6 @@ private Ints() {}
    *
    * <p>Prefer {@link Integer#BYTES} instead.
    */
-  // The constants value gets inlined here.
-  @SuppressWarnings("AndroidJdkLibsChecker")
   public static final int BYTES = Integer.BYTES;
 
   /**
diff --git a/guava/src/com/google/common/primitives/Longs.java b/guava/src/com/google/common/primitives/Longs.java
index f4ee31a73a4a..b253d45ed4f8 100644
--- a/guava/src/com/google/common/primitives/Longs.java
+++ b/guava/src/com/google/common/primitives/Longs.java
@@ -55,8 +55,6 @@ private Longs() {}
    *
    * <p>Prefer {@link Long#BYTES} instead.
    */
-  // The constants value gets inlined here.
-  @SuppressWarnings("AndroidJdkLibsChecker")
   public static final int BYTES = Long.BYTES;
 
   /**
diff --git a/guava/src/com/google/common/primitives/Shorts.java b/guava/src/com/google/common/primitives/Shorts.java
index 3a54130d8169..1e95a081b701 100644
--- a/guava/src/com/google/common/primitives/Shorts.java
+++ b/guava/src/com/google/common/primitives/Shorts.java
@@ -54,8 +54,6 @@ private Shorts() {}
    *
    * <p>Prefer {@link Short#BYTES} instead.
    */
-  // The constants value gets inlined here.
-  @SuppressWarnings("AndroidJdkLibsChecker")
   public static final int BYTES = Short.BYTES;
 
   /**
