diff --git a/android/guava/src/com/google/common/graph/BaseGraph.java b/android/guava/src/com/google/common/graph/BaseGraph.java
index 4d41f389f81a..fe83506b3dad 100644
--- a/android/guava/src/com/google/common/graph/BaseGraph.java
+++ b/android/guava/src/com/google/common/graph/BaseGraph.java
@@ -50,6 +50,8 @@ interface BaseGraph<N> extends ArchetypeGraph<N> {
    * <p>This capability facilitates writing implementations of <a
    * href="https://github.com/google/guava/wiki/GraphsExplained#graph-types-for-algorithms">edge-oriented
    * code</a>.
+   *
+   * @since NEXT
    */
   Network<N, EndpointPair<N>> asNetwork();
 
diff --git a/guava/src/com/google/common/graph/BaseGraph.java b/guava/src/com/google/common/graph/BaseGraph.java
index 4d41f389f81a..fe83506b3dad 100644
--- a/guava/src/com/google/common/graph/BaseGraph.java
+++ b/guava/src/com/google/common/graph/BaseGraph.java
@@ -50,6 +50,8 @@ interface BaseGraph<N> extends ArchetypeGraph<N> {
    * <p>This capability facilitates writing implementations of <a
    * href="https://github.com/google/guava/wiki/GraphsExplained#graph-types-for-algorithms">edge-oriented
    * code</a>.
+   *
+   * @since NEXT
    */
   Network<N, EndpointPair<N>> asNetwork();
 
