diff --git a/android/guava-tests/test/com/google/common/collect/IterablesTest.java b/android/guava-tests/test/com/google/common/collect/IterablesTest.java
index d2a6de43f7b5..71fa3928c6a0 100644
--- a/android/guava-tests/test/com/google/common/collect/IterablesTest.java
+++ b/android/guava-tests/test/com/google/common/collect/IterablesTest.java
@@ -54,6 +54,7 @@
 import java.util.Arrays;
 import java.util.Collection;
 import java.util.Collections;
+import java.util.Comparator;
 import java.util.ConcurrentModificationException;
 import java.util.HashSet;
 import java.util.Iterator;
@@ -1336,4 +1337,14 @@ private static void verifyMergeSorted(
 
     assertEquals(Lists.newLinkedList(expected), Lists.newLinkedList(mergedIterator));
   }
+
+  public void testMergeSorted_stable_allEqual() {
+    ImmutableList<Integer> first = ImmutableList.of(0, 2, 4, 6, 8);
+    ImmutableList<Integer> second = ImmutableList.of(1, 3, 5, 7, 9);
+
+    Comparator<Object> comparator = Ordering.allEqual();
+    Iterable<Integer> merged = Iterables.mergeSorted(ImmutableList.of(first, second), comparator);
+
+    assertThat(merged).containsExactly(0, 2, 4, 6, 8, 1, 3, 5, 7, 9).inOrder();
+  }
 }
diff --git a/guava-tests/test/com/google/common/collect/IterablesTest.java b/guava-tests/test/com/google/common/collect/IterablesTest.java
index 5a5b9ca55e73..03bf30d244f6 100644
--- a/guava-tests/test/com/google/common/collect/IterablesTest.java
+++ b/guava-tests/test/com/google/common/collect/IterablesTest.java
@@ -54,6 +54,7 @@
 import java.util.Arrays;
 import java.util.Collection;
 import java.util.Collections;
+import java.util.Comparator;
 import java.util.ConcurrentModificationException;
 import java.util.HashSet;
 import java.util.Iterator;
@@ -1373,4 +1374,14 @@ private static void verifyMergeSorted(
 
     assertEquals(Lists.newLinkedList(expected), Lists.newLinkedList(mergedIterator));
   }
+
+  public void testMergeSorted_stable_allEqual() {
+    ImmutableList<Integer> first = ImmutableList.of(0, 2, 4, 6, 8);
+    ImmutableList<Integer> second = ImmutableList.of(1, 3, 5, 7, 9);
+
+    Comparator<Object> comparator = Ordering.allEqual();
+    Iterable<Integer> merged = Iterables.mergeSorted(ImmutableList.of(first, second), comparator);
+
+    assertThat(merged).containsExactly(0, 2, 4, 6, 8, 1, 3, 5, 7, 9).inOrder();
+  }
 }
