diff --git a/android/guava-tests/test/com/google/common/collect/MultimapsCollectionTest.java b/android/guava-tests/test/com/google/common/collect/MultimapsCollectionTest.java
index 439aab2f7f3c..882abdd8e131 100644
--- a/android/guava-tests/test/com/google/common/collect/MultimapsCollectionTest.java
+++ b/android/guava-tests/test/com/google/common/collect/MultimapsCollectionTest.java
@@ -30,7 +30,7 @@
 import static com.google.common.collect.testing.google.MultisetElementSetTester.getElementSetDuplicateInitializingMethods;
 import static com.google.common.collect.testing.google.MultisetIteratorTester.getIteratorDuplicateInitializingMethods;
 import static com.google.common.collect.testing.google.MultisetRemoveTester.getRemoveDuplicateInitializingMethods;
-import static java.lang.reflect.Proxy.newProxyInstance;
+import static com.google.common.reflect.Reflection.newProxy;
 
 import com.google.common.annotations.GwtIncompatible;
 import com.google.common.base.Ascii;
@@ -57,8 +57,6 @@
 import com.google.common.collect.testing.google.TestSetMultimapGenerator;
 import com.google.common.collect.testing.google.TestStringListMultimapGenerator;
 import com.google.common.collect.testing.google.TestStringMultisetGenerator;
-import java.lang.reflect.InvocationHandler;
-import java.lang.reflect.Method;
 import java.util.Collection;
 import java.util.HashMap;
 import java.util.List;
@@ -150,15 +148,10 @@ static <K, V> PopulatableMapAsMultimap<K, V> create() {
       this.map = new HashMap<>();
       this.unusableDelegate =
           (SetMultimap<K, V>)
-              newProxyInstance(
-                  SetMultimap.class.getClassLoader(),
-                  new Class<?>[] {SetMultimap.class},
-                  new InvocationHandler() {
-                    @Override
-                    public Object invoke(Object proxy, Method method, Object[] args)
-                        throws Throwable {
-                      throw new UnsupportedOperationException();
-                    }
+              newProxy(
+                  SetMultimap.class,
+                  (proxy, method, args) -> {
+                    throw new UnsupportedOperationException();
                   });
     }
 
diff --git a/guava-tests/test/com/google/common/collect/MultimapsCollectionTest.java b/guava-tests/test/com/google/common/collect/MultimapsCollectionTest.java
index 43b0e93083af..6b683e13633d 100644
--- a/guava-tests/test/com/google/common/collect/MultimapsCollectionTest.java
+++ b/guava-tests/test/com/google/common/collect/MultimapsCollectionTest.java
@@ -31,7 +31,7 @@
 import static com.google.common.collect.testing.google.MultisetForEachEntryTester.getForEachEntryDuplicateInitializingMethods;
 import static com.google.common.collect.testing.google.MultisetIteratorTester.getIteratorDuplicateInitializingMethods;
 import static com.google.common.collect.testing.google.MultisetRemoveTester.getRemoveDuplicateInitializingMethods;
-import static java.lang.reflect.Proxy.newProxyInstance;
+import static com.google.common.reflect.Reflection.newProxy;
 
 import com.google.common.annotations.GwtIncompatible;
 import com.google.common.base.Ascii;
@@ -58,8 +58,6 @@
 import com.google.common.collect.testing.google.TestSetMultimapGenerator;
 import com.google.common.collect.testing.google.TestStringListMultimapGenerator;
 import com.google.common.collect.testing.google.TestStringMultisetGenerator;
-import java.lang.reflect.InvocationHandler;
-import java.lang.reflect.Method;
 import java.util.Collection;
 import java.util.HashMap;
 import java.util.List;
@@ -151,15 +149,10 @@ static <K, V> PopulatableMapAsMultimap<K, V> create() {
       this.map = new HashMap<>();
       this.unusableDelegate =
           (SetMultimap<K, V>)
-              newProxyInstance(
-                  SetMultimap.class.getClassLoader(),
-                  new Class<?>[] {SetMultimap.class},
-                  new InvocationHandler() {
-                    @Override
-                    public Object invoke(Object proxy, Method method, Object[] args)
-                        throws Throwable {
-                      throw new UnsupportedOperationException();
-                    }
+              newProxy(
+                  SetMultimap.class,
+                  (proxy, method, args) -> {
+                    throw new UnsupportedOperationException();
                   });
     }
 
