diff --git a/android/guava/src/com/google/common/collect/Maps.java b/android/guava/src/com/google/common/collect/Maps.java
index 7b29fd5714a0..1c75888c388e 100644
--- a/android/guava/src/com/google/common/collect/Maps.java
+++ b/android/guava/src/com/google/common/collect/Maps.java
@@ -531,7 +531,7 @@ MapDifference<K, V> difference(
     Map<K, V> onlyOnLeft = new LinkedHashMap<>();
     Map<K, V> onlyOnRight = new LinkedHashMap<>(right); // will whittle it down
     Map<K, V> onBoth = new LinkedHashMap<>();
-    Map<K, MapDifference.ValueDifference<V>> differences = new LinkedHashMap<>();
+    Map<K, ValueDifference<V>> differences = new LinkedHashMap<>();
     doDifference(left, right, valueEquivalence, onlyOnLeft, onlyOnRight, onBoth, differences);
     return new MapDifferenceImpl<>(onlyOnLeft, onlyOnRight, onBoth, differences);
   }
@@ -559,11 +559,11 @@ SortedMapDifference<K, V> difference(
     checkNotNull(left);
     checkNotNull(right);
     Comparator<? super K> comparator = orNaturalOrder(left.comparator());
-    SortedMap<K, V> onlyOnLeft = Maps.newTreeMap(comparator);
-    SortedMap<K, V> onlyOnRight = Maps.newTreeMap(comparator);
+    SortedMap<K, V> onlyOnLeft = newTreeMap(comparator);
+    SortedMap<K, V> onlyOnRight = newTreeMap(comparator);
     onlyOnRight.putAll(right); // will whittle it down
-    SortedMap<K, V> onBoth = Maps.newTreeMap(comparator);
-    SortedMap<K, MapDifference.ValueDifference<V>> differences = Maps.newTreeMap(comparator);
+    SortedMap<K, V> onBoth = newTreeMap(comparator);
+    SortedMap<K, ValueDifference<V>> differences = newTreeMap(comparator);
 
     doDifference(left, right, Equivalence.equals(), onlyOnLeft, onlyOnRight, onBoth, differences);
     return new SortedMapDifferenceImpl<>(onlyOnLeft, onlyOnRight, onBoth, differences);
@@ -576,7 +576,7 @@ SortedMapDifference<K, V> difference(
       Map<K, V> onlyOnLeft,
       Map<K, V> onlyOnRight,
       Map<K, V> onBoth,
-      Map<K, MapDifference.ValueDifference<V>> differences) {
+      Map<K, ValueDifference<V>> differences) {
     for (Entry<? extends K, ? extends V> entry : left.entrySet()) {
       K leftKey = entry.getKey();
       V leftValue = entry.getValue();
@@ -696,8 +696,7 @@ public String toString() {
     }
   }
 
-  static final class ValueDifferenceImpl<V extends @Nullable Object>
-      implements MapDifference.ValueDifference<V> {
+  static final class ValueDifferenceImpl<V extends @Nullable Object> implements ValueDifference<V> {
     @ParametricNullness private final V left;
     @ParametricNullness private final V right;
 
@@ -725,8 +724,8 @@ public V rightValue() {
 
     @Override
     public boolean equals(@Nullable Object object) {
-      if (object instanceof MapDifference.ValueDifference) {
-        MapDifference.ValueDifference<?> that = (MapDifference.ValueDifference<?>) object;
+      if (object instanceof ValueDifference) {
+        ValueDifference<?> that = (ValueDifference<?>) object;
         return Objects.equals(this.left, that.leftValue())
             && Objects.equals(this.right, that.rightValue());
       }
@@ -1627,8 +1626,7 @@ public String toString() {
    * of its collection views:
    *
    * {@snippet :
-   * BiMap<Long, String> map = Maps.synchronizedBiMap(
-   *     HashBiMap.<Long, String>create());
+   * BiMap<Long, String> map = Maps.synchronizedBiMap(HashBiMap.create());
    * ...
    * Set<Long> set = map.keySet();  // Needn't be in synchronized block
    * ...
@@ -2109,7 +2107,7 @@ public Set<K> keySet() {
     @Override
     Iterator<Entry<K, V2>> entryIterator() {
       return Iterators.transform(
-          fromMap.entrySet().iterator(), Maps.<K, V1, V2>asEntryToEntryFunction(transformer));
+          fromMap.entrySet().iterator(), asEntryToEntryFunction(transformer));
     }
 
     @Override
@@ -2296,12 +2294,12 @@ protected NavigableMap<K, V1> fromMap() {
 
   static <K extends @Nullable Object> Predicate<Entry<K, ?>> keyPredicateOnEntries(
       Predicate<? super K> keyPredicate) {
-    return compose(keyPredicate, Maps.<K>keyFunction());
+    return compose(keyPredicate, keyFunction());
   }
 
   static <V extends @Nullable Object> Predicate<Entry<?, V>> valuePredicateOnEntries(
       Predicate<? super V> valuePredicate) {
-    return compose(valuePredicate, Maps.<V>valueFunction());
+    return compose(valuePredicate, valueFunction());
   }
 
   /**
@@ -2364,9 +2362,8 @@ protected NavigableMap<K, V1> fromMap() {
    */
   public static <K extends @Nullable Object, V extends @Nullable Object> SortedMap<K, V> filterKeys(
       SortedMap<K, V> unfiltered, Predicate<? super K> keyPredicate) {
-    // TODO(lowasser): Return a subclass of Maps.FilteredKeyMap for slightly better
-    // performance.
-    return filterEntries(unfiltered, Maps.<K>keyPredicateOnEntries(keyPredicate));
+    // TODO(lowasser): Return a subclass of FilteredKeyMap for slightly better performance.
+    return filterEntries(unfiltered, keyPredicateOnEntries(keyPredicate));
   }
 
   /**
@@ -2399,9 +2396,8 @@ protected NavigableMap<K, V1> fromMap() {
   public static <K extends @Nullable Object, V extends @Nullable Object>
       NavigableMap<K, V> filterKeys(
           NavigableMap<K, V> unfiltered, Predicate<? super K> keyPredicate) {
-    // TODO(lowasser): Return a subclass of Maps.FilteredKeyMap for slightly better
-    // performance.
-    return filterEntries(unfiltered, Maps.<K>keyPredicateOnEntries(keyPredicate));
+    // TODO(lowasser): Return a subclass of FilteredKeyMap for slightly better performance.
+    return filterEntries(unfiltered, keyPredicateOnEntries(keyPredicate));
   }
 
   /**
@@ -2431,7 +2427,7 @@ NavigableMap<K, V> filterKeys(
   public static <K extends @Nullable Object, V extends @Nullable Object> BiMap<K, V> filterKeys(
       BiMap<K, V> unfiltered, Predicate<? super K> keyPredicate) {
     checkNotNull(keyPredicate);
-    return filterEntries(unfiltered, Maps.<K>keyPredicateOnEntries(keyPredicate));
+    return filterEntries(unfiltered, keyPredicateOnEntries(keyPredicate));
   }
 
   /**
@@ -2459,7 +2455,7 @@ NavigableMap<K, V> filterKeys(
    */
   public static <K extends @Nullable Object, V extends @Nullable Object> Map<K, V> filterValues(
       Map<K, V> unfiltered, Predicate<? super V> valuePredicate) {
-    return filterEntries(unfiltered, Maps.<V>valuePredicateOnEntries(valuePredicate));
+    return filterEntries(unfiltered, valuePredicateOnEntries(valuePredicate));
   }
 
   /**
@@ -2491,7 +2487,7 @@ NavigableMap<K, V> filterKeys(
   public static <K extends @Nullable Object, V extends @Nullable Object>
       SortedMap<K, V> filterValues(
           SortedMap<K, V> unfiltered, Predicate<? super V> valuePredicate) {
-    return filterEntries(unfiltered, Maps.<V>valuePredicateOnEntries(valuePredicate));
+    return filterEntries(unfiltered, valuePredicateOnEntries(valuePredicate));
   }
 
   /**
@@ -2524,7 +2520,7 @@ SortedMap<K, V> filterValues(
   public static <K extends @Nullable Object, V extends @Nullable Object>
       NavigableMap<K, V> filterValues(
           NavigableMap<K, V> unfiltered, Predicate<? super V> valuePredicate) {
-    return filterEntries(unfiltered, Maps.<V>valuePredicateOnEntries(valuePredicate));
+    return filterEntries(unfiltered, valuePredicateOnEntries(valuePredicate));
   }
 
   /**
@@ -2556,7 +2552,7 @@ NavigableMap<K, V> filterValues(
    */
   public static <K extends @Nullable Object, V extends @Nullable Object> BiMap<K, V> filterValues(
       BiMap<K, V> unfiltered, Predicate<? super V> valuePredicate) {
-    return filterEntries(unfiltered, Maps.<V>valuePredicateOnEntries(valuePredicate));
+    return filterEntries(unfiltered, valuePredicateOnEntries(valuePredicate));
   }
 
   /**
@@ -2704,8 +2700,7 @@ NavigableMap<K, V> filterEntries(
    */
   private static <K extends @Nullable Object, V extends @Nullable Object> Map<K, V> filterFiltered(
       AbstractFilteredMap<K, V> map, Predicate<? super Entry<K, V>> entryPredicate) {
-    return new FilteredEntryMap<>(
-        map.unfiltered, Predicates.<Entry<K, V>>and(map.predicate, entryPredicate));
+    return new FilteredEntryMap<>(map.unfiltered, Predicates.and(map.predicate, entryPredicate));
   }
 
   /**
@@ -2715,7 +2710,7 @@ NavigableMap<K, V> filterEntries(
   private static <K extends @Nullable Object, V extends @Nullable Object>
       SortedMap<K, V> filterFiltered(
           FilteredEntrySortedMap<K, V> map, Predicate<? super Entry<K, V>> entryPredicate) {
-    Predicate<Entry<K, V>> predicate = Predicates.<Entry<K, V>>and(map.predicate, entryPredicate);
+    Predicate<Entry<K, V>> predicate = Predicates.and(map.predicate, entryPredicate);
     return new FilteredEntrySortedMap<>(map.sortedMap(), predicate);
   }
 
@@ -2727,8 +2722,7 @@ SortedMap<K, V> filterFiltered(
   private static <K extends @Nullable Object, V extends @Nullable Object>
       NavigableMap<K, V> filterFiltered(
           FilteredEntryNavigableMap<K, V> map, Predicate<? super Entry<K, V>> entryPredicate) {
-    Predicate<Entry<K, V>> predicate =
-        Predicates.<Entry<K, V>>and(map.entryPredicate, entryPredicate);
+    Predicate<Entry<K, V>> predicate = Predicates.and(map.entryPredicate, entryPredicate);
     return new FilteredEntryNavigableMap<>(map.unfiltered, predicate);
   }
 
@@ -2739,7 +2733,7 @@ NavigableMap<K, V> filterFiltered(
   private static <K extends @Nullable Object, V extends @Nullable Object>
       BiMap<K, V> filterFiltered(
           FilteredEntryBiMap<K, V> map, Predicate<? super Entry<K, V>> entryPredicate) {
-    Predicate<Entry<K, V>> predicate = Predicates.<Entry<K, V>>and(map.predicate, entryPredicate);
+    Predicate<Entry<K, V>> predicate = Predicates.and(map.predicate, entryPredicate);
     return new FilteredEntryBiMap<>(map.unfiltered(), predicate);
   }
 
@@ -2805,7 +2799,7 @@ Collection<V> createValues() {
 
   private static final class FilteredMapValues<
           K extends @Nullable Object, V extends @Nullable Object>
-      extends Maps.Values<K, V> {
+      extends Values<K, V> {
     final Map<K, V> unfiltered;
     final Predicate<? super Entry<K, V>> predicate;
 
@@ -3151,7 +3145,7 @@ private static final class FilteredEntryNavigableMap<
 
     @Override
     public NavigableSet<K> navigableKeySet() {
-      return new Maps.NavigableKeySet<K, V>(this) {
+      return new NavigableKeySet<K, V>(this) {
         @Override
         public boolean removeAll(Collection<?> collection) {
           return FilteredEntryMap.removeAllKeys(unfiltered, entryPredicate, collection);
@@ -3336,7 +3330,7 @@ NavigableMap<K, V> unmodifiableNavigableMap(NavigableMap<K, ? extends V> map) {
 
   private static <K extends @Nullable Object, V extends @Nullable Object>
       @Nullable Entry<K, V> unmodifiableOrNull(@Nullable Entry<K, ? extends V> entry) {
-    return (entry == null) ? null : Maps.unmodifiableEntry(entry);
+    return (entry == null) ? null : unmodifiableEntry(entry);
   }
 
   @GwtIncompatible // NavigableMap
@@ -3456,8 +3450,7 @@ public NavigableMap<K, V> subMap(
         boolean fromInclusive,
         @ParametricNullness K toKey,
         boolean toInclusive) {
-      return Maps.unmodifiableNavigableMap(
-          delegate.subMap(fromKey, fromInclusive, toKey, toInclusive));
+      return unmodifiableNavigableMap(delegate.subMap(fromKey, fromInclusive, toKey, toInclusive));
     }
 
     @Override
@@ -3467,7 +3460,7 @@ public SortedMap<K, V> headMap(@ParametricNullness K toKey) {
 
     @Override
     public NavigableMap<K, V> headMap(@ParametricNullness K toKey, boolean inclusive) {
-      return Maps.unmodifiableNavigableMap(delegate.headMap(toKey, inclusive));
+      return unmodifiableNavigableMap(delegate.headMap(toKey, inclusive));
     }
 
     @Override
@@ -3477,7 +3470,7 @@ public SortedMap<K, V> tailMap(@ParametricNullness K fromKey) {
 
     @Override
     public NavigableMap<K, V> tailMap(@ParametricNullness K fromKey, boolean inclusive) {
-      return Maps.unmodifiableNavigableMap(delegate.tailMap(fromKey, inclusive));
+      return unmodifiableNavigableMap(delegate.tailMap(fromKey, inclusive));
     }
   }
 
@@ -4021,7 +4014,7 @@ public boolean contains(@Nullable Object o) {
       if (o instanceof Entry) {
         Entry<?, ?> entry = (Entry<?, ?>) o;
         Object key = entry.getKey();
-        V value = Maps.safeGet(map(), key);
+        V value = safeGet(map(), key);
         return Objects.equals(value, entry.getValue()) && (value != null || map().containsKey(key));
       }
       return false;
diff --git a/guava/src/com/google/common/collect/Maps.java b/guava/src/com/google/common/collect/Maps.java
index bfaf6bd0c323..d26913393048 100644
--- a/guava/src/com/google/common/collect/Maps.java
+++ b/guava/src/com/google/common/collect/Maps.java
@@ -534,7 +534,7 @@ MapDifference<K, V> difference(
     Map<K, V> onlyOnLeft = new LinkedHashMap<>();
     Map<K, V> onlyOnRight = new LinkedHashMap<>(right); // will whittle it down
     Map<K, V> onBoth = new LinkedHashMap<>();
-    Map<K, MapDifference.ValueDifference<V>> differences = new LinkedHashMap<>();
+    Map<K, ValueDifference<V>> differences = new LinkedHashMap<>();
     doDifference(left, right, valueEquivalence, onlyOnLeft, onlyOnRight, onBoth, differences);
     return new MapDifferenceImpl<>(onlyOnLeft, onlyOnRight, onBoth, differences);
   }
@@ -562,11 +562,11 @@ SortedMapDifference<K, V> difference(
     checkNotNull(left);
     checkNotNull(right);
     Comparator<? super K> comparator = orNaturalOrder(left.comparator());
-    SortedMap<K, V> onlyOnLeft = Maps.newTreeMap(comparator);
-    SortedMap<K, V> onlyOnRight = Maps.newTreeMap(comparator);
+    SortedMap<K, V> onlyOnLeft = newTreeMap(comparator);
+    SortedMap<K, V> onlyOnRight = newTreeMap(comparator);
     onlyOnRight.putAll(right); // will whittle it down
-    SortedMap<K, V> onBoth = Maps.newTreeMap(comparator);
-    SortedMap<K, MapDifference.ValueDifference<V>> differences = Maps.newTreeMap(comparator);
+    SortedMap<K, V> onBoth = newTreeMap(comparator);
+    SortedMap<K, ValueDifference<V>> differences = newTreeMap(comparator);
 
     doDifference(left, right, Equivalence.equals(), onlyOnLeft, onlyOnRight, onBoth, differences);
     return new SortedMapDifferenceImpl<>(onlyOnLeft, onlyOnRight, onBoth, differences);
@@ -579,7 +579,7 @@ SortedMapDifference<K, V> difference(
       Map<K, V> onlyOnLeft,
       Map<K, V> onlyOnRight,
       Map<K, V> onBoth,
-      Map<K, MapDifference.ValueDifference<V>> differences) {
+      Map<K, ValueDifference<V>> differences) {
     for (Entry<? extends K, ? extends V> entry : left.entrySet()) {
       K leftKey = entry.getKey();
       V leftValue = entry.getValue();
@@ -699,8 +699,7 @@ public String toString() {
     }
   }
 
-  static final class ValueDifferenceImpl<V extends @Nullable Object>
-      implements MapDifference.ValueDifference<V> {
+  static final class ValueDifferenceImpl<V extends @Nullable Object> implements ValueDifference<V> {
     @ParametricNullness private final V left;
     @ParametricNullness private final V right;
 
@@ -728,8 +727,8 @@ public V rightValue() {
 
     @Override
     public boolean equals(@Nullable Object object) {
-      if (object instanceof MapDifference.ValueDifference) {
-        MapDifference.ValueDifference<?> that = (MapDifference.ValueDifference<?>) object;
+      if (object instanceof ValueDifference) {
+        ValueDifference<?> that = (ValueDifference<?>) object;
         return Objects.equals(this.left, that.leftValue())
             && Objects.equals(this.right, that.rightValue());
       }
@@ -1657,8 +1656,7 @@ public String toString() {
    * of its collection views:
    *
    * {@snippet :
-   * BiMap<Long, String> map = Maps.synchronizedBiMap(
-   *     HashBiMap.<Long, String>create());
+   * BiMap<Long, String> map = Maps.synchronizedBiMap(HashBiMap.create());
    * ...
    * Set<Long> set = map.keySet();  // Needn't be in synchronized block
    * ...
@@ -2197,13 +2195,13 @@ public Set<K> keySet() {
     @Override
     Iterator<Entry<K, V2>> entryIterator() {
       return Iterators.transform(
-          fromMap.entrySet().iterator(), Maps.<K, V1, V2>asEntryToEntryFunction(transformer));
+          fromMap.entrySet().iterator(), asEntryToEntryFunction(transformer));
     }
 
     @Override
     Spliterator<Entry<K, V2>> entrySpliterator() {
       return CollectSpliterators.map(
-          fromMap.entrySet().spliterator(), Maps.<K, V1, V2>asEntryToEntryFunction(transformer));
+          fromMap.entrySet().spliterator(), asEntryToEntryFunction(transformer));
     }
 
     @Override
@@ -2397,12 +2395,12 @@ protected NavigableMap<K, V1> fromMap() {
 
   static <K extends @Nullable Object> Predicate<Entry<K, ?>> keyPredicateOnEntries(
       Predicate<? super K> keyPredicate) {
-    return compose(keyPredicate, Maps.<K>keyFunction());
+    return compose(keyPredicate, keyFunction());
   }
 
   static <V extends @Nullable Object> Predicate<Entry<?, V>> valuePredicateOnEntries(
       Predicate<? super V> valuePredicate) {
-    return compose(valuePredicate, Maps.<V>valueFunction());
+    return compose(valuePredicate, valueFunction());
   }
 
   /**
@@ -2465,9 +2463,8 @@ protected NavigableMap<K, V1> fromMap() {
    */
   public static <K extends @Nullable Object, V extends @Nullable Object> SortedMap<K, V> filterKeys(
       SortedMap<K, V> unfiltered, Predicate<? super K> keyPredicate) {
-    // TODO(lowasser): Return a subclass of Maps.FilteredKeyMap for slightly better
-    // performance.
-    return filterEntries(unfiltered, Maps.<K>keyPredicateOnEntries(keyPredicate));
+    // TODO(lowasser): Return a subclass of FilteredKeyMap for slightly better performance.
+    return filterEntries(unfiltered, keyPredicateOnEntries(keyPredicate));
   }
 
   /**
@@ -2500,9 +2497,8 @@ protected NavigableMap<K, V1> fromMap() {
   public static <K extends @Nullable Object, V extends @Nullable Object>
       NavigableMap<K, V> filterKeys(
           NavigableMap<K, V> unfiltered, Predicate<? super K> keyPredicate) {
-    // TODO(lowasser): Return a subclass of Maps.FilteredKeyMap for slightly better
-    // performance.
-    return filterEntries(unfiltered, Maps.<K>keyPredicateOnEntries(keyPredicate));
+    // TODO(lowasser): Return a subclass of FilteredKeyMap for slightly better performance.
+    return filterEntries(unfiltered, keyPredicateOnEntries(keyPredicate));
   }
 
   /**
@@ -2532,7 +2528,7 @@ NavigableMap<K, V> filterKeys(
   public static <K extends @Nullable Object, V extends @Nullable Object> BiMap<K, V> filterKeys(
       BiMap<K, V> unfiltered, Predicate<? super K> keyPredicate) {
     checkNotNull(keyPredicate);
-    return filterEntries(unfiltered, Maps.<K>keyPredicateOnEntries(keyPredicate));
+    return filterEntries(unfiltered, keyPredicateOnEntries(keyPredicate));
   }
 
   /**
@@ -2560,7 +2556,7 @@ NavigableMap<K, V> filterKeys(
    */
   public static <K extends @Nullable Object, V extends @Nullable Object> Map<K, V> filterValues(
       Map<K, V> unfiltered, Predicate<? super V> valuePredicate) {
-    return filterEntries(unfiltered, Maps.<V>valuePredicateOnEntries(valuePredicate));
+    return filterEntries(unfiltered, valuePredicateOnEntries(valuePredicate));
   }
 
   /**
@@ -2592,7 +2588,7 @@ NavigableMap<K, V> filterKeys(
   public static <K extends @Nullable Object, V extends @Nullable Object>
       SortedMap<K, V> filterValues(
           SortedMap<K, V> unfiltered, Predicate<? super V> valuePredicate) {
-    return filterEntries(unfiltered, Maps.<V>valuePredicateOnEntries(valuePredicate));
+    return filterEntries(unfiltered, valuePredicateOnEntries(valuePredicate));
   }
 
   /**
@@ -2625,7 +2621,7 @@ SortedMap<K, V> filterValues(
   public static <K extends @Nullable Object, V extends @Nullable Object>
       NavigableMap<K, V> filterValues(
           NavigableMap<K, V> unfiltered, Predicate<? super V> valuePredicate) {
-    return filterEntries(unfiltered, Maps.<V>valuePredicateOnEntries(valuePredicate));
+    return filterEntries(unfiltered, valuePredicateOnEntries(valuePredicate));
   }
 
   /**
@@ -2657,7 +2653,7 @@ NavigableMap<K, V> filterValues(
    */
   public static <K extends @Nullable Object, V extends @Nullable Object> BiMap<K, V> filterValues(
       BiMap<K, V> unfiltered, Predicate<? super V> valuePredicate) {
-    return filterEntries(unfiltered, Maps.<V>valuePredicateOnEntries(valuePredicate));
+    return filterEntries(unfiltered, valuePredicateOnEntries(valuePredicate));
   }
 
   /**
@@ -2805,8 +2801,7 @@ NavigableMap<K, V> filterEntries(
    */
   private static <K extends @Nullable Object, V extends @Nullable Object> Map<K, V> filterFiltered(
       AbstractFilteredMap<K, V> map, Predicate<? super Entry<K, V>> entryPredicate) {
-    return new FilteredEntryMap<>(
-        map.unfiltered, Predicates.<Entry<K, V>>and(map.predicate, entryPredicate));
+    return new FilteredEntryMap<>(map.unfiltered, Predicates.and(map.predicate, entryPredicate));
   }
 
   /**
@@ -2816,7 +2811,7 @@ NavigableMap<K, V> filterEntries(
   private static <K extends @Nullable Object, V extends @Nullable Object>
       SortedMap<K, V> filterFiltered(
           FilteredEntrySortedMap<K, V> map, Predicate<? super Entry<K, V>> entryPredicate) {
-    Predicate<Entry<K, V>> predicate = Predicates.<Entry<K, V>>and(map.predicate, entryPredicate);
+    Predicate<Entry<K, V>> predicate = Predicates.and(map.predicate, entryPredicate);
     return new FilteredEntrySortedMap<>(map.sortedMap(), predicate);
   }
 
@@ -2828,8 +2823,7 @@ SortedMap<K, V> filterFiltered(
   private static <K extends @Nullable Object, V extends @Nullable Object>
       NavigableMap<K, V> filterFiltered(
           FilteredEntryNavigableMap<K, V> map, Predicate<? super Entry<K, V>> entryPredicate) {
-    Predicate<Entry<K, V>> predicate =
-        Predicates.<Entry<K, V>>and(map.entryPredicate, entryPredicate);
+    Predicate<Entry<K, V>> predicate = Predicates.and(map.entryPredicate, entryPredicate);
     return new FilteredEntryNavigableMap<>(map.unfiltered, predicate);
   }
 
@@ -2840,7 +2834,7 @@ NavigableMap<K, V> filterFiltered(
   private static <K extends @Nullable Object, V extends @Nullable Object>
       BiMap<K, V> filterFiltered(
           FilteredEntryBiMap<K, V> map, Predicate<? super Entry<K, V>> entryPredicate) {
-    Predicate<Entry<K, V>> predicate = Predicates.<Entry<K, V>>and(map.predicate, entryPredicate);
+    Predicate<Entry<K, V>> predicate = Predicates.and(map.predicate, entryPredicate);
     return new FilteredEntryBiMap<>(map.unfiltered(), predicate);
   }
 
@@ -2906,7 +2900,7 @@ Collection<V> createValues() {
 
   private static final class FilteredMapValues<
           K extends @Nullable Object, V extends @Nullable Object>
-      extends Maps.Values<K, V> {
+      extends Values<K, V> {
     final Map<K, V> unfiltered;
     final Predicate<? super Entry<K, V>> predicate;
 
@@ -3252,7 +3246,7 @@ private static final class FilteredEntryNavigableMap<
 
     @Override
     public NavigableSet<K> navigableKeySet() {
-      return new Maps.NavigableKeySet<K, V>(this) {
+      return new NavigableKeySet<K, V>(this) {
         @Override
         public boolean removeAll(Collection<?> collection) {
           return FilteredEntryMap.removeAllKeys(unfiltered, entryPredicate, collection);
@@ -3447,7 +3441,7 @@ NavigableMap<K, V> unmodifiableNavigableMap(NavigableMap<K, ? extends V> map) {
 
   private static <K extends @Nullable Object, V extends @Nullable Object>
       @Nullable Entry<K, V> unmodifiableOrNull(@Nullable Entry<K, ? extends V> entry) {
-    return (entry == null) ? null : Maps.unmodifiableEntry(entry);
+    return (entry == null) ? null : unmodifiableEntry(entry);
   }
 
   @GwtIncompatible // NavigableMap
@@ -3619,8 +3613,7 @@ public NavigableMap<K, V> subMap(
         boolean fromInclusive,
         @ParametricNullness K toKey,
         boolean toInclusive) {
-      return Maps.unmodifiableNavigableMap(
-          delegate.subMap(fromKey, fromInclusive, toKey, toInclusive));
+      return unmodifiableNavigableMap(delegate.subMap(fromKey, fromInclusive, toKey, toInclusive));
     }
 
     @Override
@@ -3630,7 +3623,7 @@ public SortedMap<K, V> headMap(@ParametricNullness K toKey) {
 
     @Override
     public NavigableMap<K, V> headMap(@ParametricNullness K toKey, boolean inclusive) {
-      return Maps.unmodifiableNavigableMap(delegate.headMap(toKey, inclusive));
+      return unmodifiableNavigableMap(delegate.headMap(toKey, inclusive));
     }
 
     @Override
@@ -3640,7 +3633,7 @@ public SortedMap<K, V> tailMap(@ParametricNullness K fromKey) {
 
     @Override
     public NavigableMap<K, V> tailMap(@ParametricNullness K fromKey, boolean inclusive) {
-      return Maps.unmodifiableNavigableMap(delegate.tailMap(fromKey, inclusive));
+      return unmodifiableNavigableMap(delegate.tailMap(fromKey, inclusive));
     }
   }
 
@@ -4217,7 +4210,7 @@ public boolean contains(@Nullable Object o) {
       if (o instanceof Entry) {
         Entry<?, ?> entry = (Entry<?, ?>) o;
         Object key = entry.getKey();
-        V value = Maps.safeGet(map(), key);
+        V value = safeGet(map(), key);
         return Objects.equals(value, entry.getValue()) && (value != null || map().containsKey(key));
       }
       return false;
