diff --git a/android/guava/src/com/google/common/collect/Streams.java b/android/guava/src/com/google/common/collect/Streams.java
index 293ab0b570ca..c58895f5c4dd 100644
--- a/android/guava/src/com/google/common/collect/Streams.java
+++ b/android/guava/src/com/google/common/collect/Streams.java
@@ -531,7 +531,6 @@ Splitr createSplit(Spliterator<T> from, long i) {
    * <p>The order of the resulting stream is defined if and only if the order of the original stream
    * was defined.
    */
-  @SuppressWarnings("AndroidJdkLibsChecker") // b/229998664
   public static <R extends @Nullable Object> Stream<R> mapWithIndex(
       IntStream stream, IntFunctionWithIndex<R> function) {
     checkNotNull(stream);
@@ -611,7 +610,6 @@ Splitr createSplit(Spliterator.OfInt from, long i) {
    * <p>The order of the resulting stream is defined if and only if the order of the original stream
    * was defined.
    */
-  @SuppressWarnings("AndroidJdkLibsChecker") // b/229998664
   public static <R extends @Nullable Object> Stream<R> mapWithIndex(
       LongStream stream, LongFunctionWithIndex<R> function) {
     checkNotNull(stream);
@@ -691,7 +689,6 @@ Splitr createSplit(Spliterator.OfLong from, long i) {
    * <p>The order of the resulting stream is defined if and only if the order of the original stream
    * was defined.
    */
-  @SuppressWarnings("AndroidJdkLibsChecker") // b/229998664
   public static <R extends @Nullable Object> Stream<R> mapWithIndex(
       DoubleStream stream, DoubleFunctionWithIndex<R> function) {
     checkNotNull(stream);
diff --git a/android/guava/src/com/google/common/reflect/Types.java b/android/guava/src/com/google/common/reflect/Types.java
index 3f863910df57..faef3d8ad4b3 100644
--- a/android/guava/src/com/google/common/reflect/Types.java
+++ b/android/guava/src/com/google/common/reflect/Types.java
@@ -627,7 +627,6 @@ Type usedInGenericType(Type type) {
       }
 
       @IgnoreJRERequirement
-      @SuppressWarnings("AndroidJdkLibsChecker") // as discussed under JAVA8.typeName
       @Override
       String typeName(Type type) {
         return type.getTypeName();
diff --git a/guava/src/com/google/common/reflect/Types.java b/guava/src/com/google/common/reflect/Types.java
index 74718cb813cd..1f8810f9cdc3 100644
--- a/guava/src/com/google/common/reflect/Types.java
+++ b/guava/src/com/google/common/reflect/Types.java
@@ -625,7 +625,6 @@ Type usedInGenericType(Type type) {
         return JAVA8.usedInGenericType(type);
       }
 
-      @SuppressWarnings("AndroidJdkLibsChecker") // as discussed under JAVA8.typeName
       @Override
       String typeName(Type type) {
         return type.getTypeName();
