diff --git a/android/guava/src/com/google/common/io/BaseEncoding.java b/android/guava/src/com/google/common/io/BaseEncoding.java
index 2de70839c007..a12d7519f55f 100644
--- a/android/guava/src/com/google/common/io/BaseEncoding.java
+++ b/android/guava/src/com/google/common/io/BaseEncoding.java
@@ -425,6 +425,8 @@ public static BaseEncoding base32Hex() {
    * <p>No line feeds are added by default, as per <a
    * href="http://tools.ietf.org/html/rfc4648#section-3.1">RFC 4648 section 3.1</a>, Line Feeds in
    * Encoded Data. Line feeds may be added using {@link #withSeparator(String, int)}.
+   *
+   * <p><b>Java 17+ users:</b> Consider using the {@link java.util.HexFormat} API instead.
    */
   public static BaseEncoding base16() {
     return BASE16;
diff --git a/guava/src/com/google/common/io/BaseEncoding.java b/guava/src/com/google/common/io/BaseEncoding.java
index 2de70839c007..a12d7519f55f 100644
--- a/guava/src/com/google/common/io/BaseEncoding.java
+++ b/guava/src/com/google/common/io/BaseEncoding.java
@@ -425,6 +425,8 @@ public static BaseEncoding base32Hex() {
    * <p>No line feeds are added by default, as per <a
    * href="http://tools.ietf.org/html/rfc4648#section-3.1">RFC 4648 section 3.1</a>, Line Feeds in
    * Encoded Data. Line feeds may be added using {@link #withSeparator(String, int)}.
+   *
+   * <p><b>Java 17+ users:</b> Consider using the {@link java.util.HexFormat} API instead.
    */
   public static BaseEncoding base16() {
     return BASE16;
