diff --git a/README.md b/README.md
index 85cc4cf3156a..9b19e05298f0 100644
--- a/README.md
+++ b/README.md
@@ -28,8 +28,8 @@ Guava comes in two flavors:
 Guava's Maven group ID is `com.google.guava`, and its artifact ID is `guava`.
 Guava provides two different "flavors": one for use on a (Java 8+) JRE and one
 for use on Android or by any library that wants to be compatible with Android.
-These flavors are specified in the Maven version field as either `33.4.8-jre` or
-`33.4.8-android`. For more about depending on Guava, see
+These flavors are specified in the Maven version field as either `33.5.0-jre` or
+`33.5.0-android`. For more about depending on Guava, see
 [using Guava in your build].
 
 To add a dependency on Guava using Maven, use the following:
@@ -38,9 +38,9 @@ To add a dependency on Guava using Maven, use the following:
 <dependency>
   <groupId>com.google.guava</groupId>
   <artifactId>guava</artifactId>
-  <version>33.4.8-jre</version>
+  <version>33.5.0-jre</version>
   <!-- or, for Android: -->
-  <version>33.4.8-android</version>
+  <version>33.5.0-android</version>
 </dependency>
 ```
 
@@ -51,16 +51,16 @@ dependencies {
   // Pick one:
 
   // 1. Use Guava in your implementation only:
-  implementation("com.google.guava:guava:33.4.8-jre")
+  implementation("com.google.guava:guava:33.5.0-jre")
 
   // 2. Use Guava types in your public API:
-  api("com.google.guava:guava:33.4.8-jre")
+  api("com.google.guava:guava:33.5.0-jre")
 
   // 3. Android - Use Guava in your implementation only:
-  implementation("com.google.guava:guava:33.4.8-android")
+  implementation("com.google.guava:guava:33.5.0-android")
 
   // 4. Android - Use Guava types in your public API:
-  api("com.google.guava:guava:33.4.8-android")
+  api("com.google.guava:guava:33.5.0-android")
 }
 ```
 
diff --git a/android/guava-testlib/src/com/google/common/testing/CollectorTester.java b/android/guava-testlib/src/com/google/common/testing/CollectorTester.java
index e54524dfe952..263d5b661ef6 100644
--- a/android/guava-testlib/src/com/google/common/testing/CollectorTester.java
+++ b/android/guava-testlib/src/com/google/common/testing/CollectorTester.java
@@ -44,7 +44,7 @@
  * </pre>
  *
  * @author Louis Wasserman
- * @since NEXT (but since 21.0 in the JRE flavor)
+ * @since 33.5.0 (but since 21.0 in the JRE flavor)
  */
 @GwtCompatible
 @NullMarked
diff --git a/android/guava/src/com/google/common/math/IntMath.java b/android/guava/src/com/google/common/math/IntMath.java
index bdd6c471eb71..a959dc4968a7 100644
--- a/android/guava/src/com/google/common/math/IntMath.java
+++ b/android/guava/src/com/google/common/math/IntMath.java
@@ -751,7 +751,7 @@ public static boolean isPrime(int n) {
    * hash code) into a well-distributed nonnegative number, the most even distribution is achieved
    * not by this method or other absolute value methods, but by {@code x & Integer.MAX_VALUE}.
    *
-   * @since NEXT
+   * @since 33.5.0
    */
   public static int saturatedAbs(int x) {
     return (x == Integer.MIN_VALUE) ? Integer.MAX_VALUE : Math.abs(x);
diff --git a/android/guava/src/com/google/common/math/LongMath.java b/android/guava/src/com/google/common/math/LongMath.java
index c1c91cdf9732..4e26183d658f 100644
--- a/android/guava/src/com/google/common/math/LongMath.java
+++ b/android/guava/src/com/google/common/math/LongMath.java
@@ -1360,7 +1360,7 @@ public static double roundToDouble(long x, RoundingMode mode) {
    * into a well-distributed nonnegative number, the most even distribution is achieved not by this
    * method or other absolute value methods, but by {@code x & Long.MAX_VALUE}.
    *
-   * @since NEXT
+   * @since 33.5.0
    */
   public static long saturatedAbs(long x) {
     return (x == Long.MIN_VALUE) ? Long.MAX_VALUE : Math.abs(x);
diff --git a/android/guava/src/com/google/common/net/MediaType.java b/android/guava/src/com/google/common/net/MediaType.java
index 4cd592bea1eb..d88c9dd684ed 100644
--- a/android/guava/src/com/google/common/net/MediaType.java
+++ b/android/guava/src/com/google/common/net/MediaType.java
@@ -256,7 +256,7 @@ private static MediaType addKnownType(MediaType mediaType) {
   /**
    * <a href="https://en.wikipedia.org/wiki/AVIF">AVIF image format</a>.
    *
-   * @since NEXT
+   * @since 33.5.0
    */
   public static final MediaType AVIF = createConstant(IMAGE_TYPE, "avif");
 
diff --git a/android/guava/src/com/google/common/util/concurrent/Striped.java b/android/guava/src/com/google/common/util/concurrent/Striped.java
index c98eb57c32b1..110a62ba8759 100644
--- a/android/guava/src/com/google/common/util/concurrent/Striped.java
+++ b/android/guava/src/com/google/common/util/concurrent/Striped.java
@@ -192,7 +192,7 @@ public Iterable<L> bulkGet(Iterable<? extends Object> keys) {
    * @param stripes the minimum number of stripes (locks) required
    * @param supplier a {@code Supplier<L>} object to obtain locks from
    * @return a new {@code Striped<L>}
-   * @since NEXT
+   * @since 33.5.0
    */
   public static <L> Striped<L> custom(int stripes, Supplier<L> supplier) {
     return new CompactStriped<>(stripes, supplier);
diff --git a/guava-testlib/README.md b/guava-testlib/README.md
index f4f26f42e60c..a9ed85ffb3cc 100644
--- a/guava-testlib/README.md
+++ b/guava-testlib/README.md
@@ -13,7 +13,7 @@ To add a dependency on Guava testlib using Maven, use the following:
 <dependency>
   <groupId>com.google.guava</groupId>
   <artifactId>guava-testlib</artifactId>
-  <version>33.4.8-jre</version>
+  <version>33.5.0-jre</version>
   <scope>test</scope>
 </dependency>
 ```
@@ -22,7 +22,7 @@ To add a dependency using Gradle:
 
 ```gradle
 dependencies {
-  test 'com.google.guava:guava-testlib:33.4.8-jre'
+  test 'com.google.guava:guava-testlib:33.5.0-jre'
 }
 ```
 
diff --git a/guava/src/com/google/common/math/IntMath.java b/guava/src/com/google/common/math/IntMath.java
index bdd6c471eb71..a959dc4968a7 100644
--- a/guava/src/com/google/common/math/IntMath.java
+++ b/guava/src/com/google/common/math/IntMath.java
@@ -751,7 +751,7 @@ public static boolean isPrime(int n) {
    * hash code) into a well-distributed nonnegative number, the most even distribution is achieved
    * not by this method or other absolute value methods, but by {@code x & Integer.MAX_VALUE}.
    *
-   * @since NEXT
+   * @since 33.5.0
    */
   public static int saturatedAbs(int x) {
     return (x == Integer.MIN_VALUE) ? Integer.MAX_VALUE : Math.abs(x);
diff --git a/guava/src/com/google/common/math/LongMath.java b/guava/src/com/google/common/math/LongMath.java
index c1c91cdf9732..4e26183d658f 100644
--- a/guava/src/com/google/common/math/LongMath.java
+++ b/guava/src/com/google/common/math/LongMath.java
@@ -1360,7 +1360,7 @@ public static double roundToDouble(long x, RoundingMode mode) {
    * into a well-distributed nonnegative number, the most even distribution is achieved not by this
    * method or other absolute value methods, but by {@code x & Long.MAX_VALUE}.
    *
-   * @since NEXT
+   * @since 33.5.0
    */
   public static long saturatedAbs(long x) {
     return (x == Long.MIN_VALUE) ? Long.MAX_VALUE : Math.abs(x);
diff --git a/guava/src/com/google/common/net/MediaType.java b/guava/src/com/google/common/net/MediaType.java
index 4cd592bea1eb..d88c9dd684ed 100644
--- a/guava/src/com/google/common/net/MediaType.java
+++ b/guava/src/com/google/common/net/MediaType.java
@@ -256,7 +256,7 @@ private static MediaType addKnownType(MediaType mediaType) {
   /**
    * <a href="https://en.wikipedia.org/wiki/AVIF">AVIF image format</a>.
    *
-   * @since NEXT
+   * @since 33.5.0
    */
   public static final MediaType AVIF = createConstant(IMAGE_TYPE, "avif");
 
diff --git a/guava/src/com/google/common/util/concurrent/Striped.java b/guava/src/com/google/common/util/concurrent/Striped.java
index c98eb57c32b1..110a62ba8759 100644
--- a/guava/src/com/google/common/util/concurrent/Striped.java
+++ b/guava/src/com/google/common/util/concurrent/Striped.java
@@ -192,7 +192,7 @@ public Iterable<L> bulkGet(Iterable<? extends Object> keys) {
    * @param stripes the minimum number of stripes (locks) required
    * @param supplier a {@code Supplier<L>} object to obtain locks from
    * @return a new {@code Striped<L>}
-   * @since NEXT
+   * @since 33.5.0
    */
   public static <L> Striped<L> custom(int stripes, Supplier<L> supplier) {
     return new CompactStriped<>(stripes, supplier);
