diff --git a/guava/src/com/google/common/collect/ImmutableSet.java b/guava/src/com/google/common/collect/ImmutableSet.java
index 5aae4e9977ee..7a6ba99d14d5 100644
--- a/guava/src/com/google/common/collect/ImmutableSet.java
+++ b/guava/src/com/google/common/collect/ImmutableSet.java
@@ -270,9 +270,9 @@ private static <E> ImmutableSet<E> fromArrayWithExpectedSize(E[] elements, int e
     }
   }
 
-  @SuppressWarnings({"rawtypes", "unchecked"}) // necessary to compile against Java 8
+  @SuppressWarnings("rawtypes") // necessary to compile against Java 8
   private static ImmutableSet copyOfEnumSet(EnumSet<?> enumSet) {
-    return ImmutableEnumSet.asImmutable(EnumSet.copyOf((EnumSet) enumSet));
+    return ImmutableEnumSet.asImmutable(EnumSet.copyOf(enumSet));
   }
 
   ImmutableSet() {}
