diff --git a/libs/exponential-histogram/src/main/java/org/elasticsearch/exponentialhistogram/CompressedHistogramData.java b/libs/exponential-histogram/src/main/java/org/elasticsearch/exponentialhistogram/CompressedHistogramData.java
index 8c80a6578f736..1bc48fad5e720 100644
--- a/libs/exponential-histogram/src/main/java/org/elasticsearch/exponentialhistogram/CompressedHistogramData.java
+++ b/libs/exponential-histogram/src/main/java/org/elasticsearch/exponentialhistogram/CompressedHistogramData.java
@@ -117,11 +117,10 @@ static void write(OutputStream output, int scale, BucketIterator negativeBuckets
         }
         output.write((byte) scaleWithFlags);
         if (hasNegativeBuckets) {
-            ByteArrayOutputStream temp = new ByteArrayOutputStream();
+            AccessibleByteArrayOutputStream temp = new AccessibleByteArrayOutputStream();
             BucketsDecoder.serializeBuckets(temp, negativeBuckets);
-            byte[] data = temp.toByteArray();
-            writeVLong(data.length, output);
-            output.write(data);
+            writeVLong(temp.size(), output);
+            output.write(temp.getBufferDirect(), 0, temp.size());
         }
         BucketsDecoder.serializeBuckets(output, positiveBuckets);
     }
@@ -252,6 +251,14 @@ private static void serializeBuckets(OutputStream out, BucketIterator buckets) t
         }
     }
 
+    private static class AccessibleByteArrayOutputStream extends ByteArrayOutputStream {
+
+        byte[] getBufferDirect() {
+            return buf;
+        }
+
+    }
+
     private static class AccessibleByteArrayStreamInput extends ByteArrayInputStream {
 
         AccessibleByteArrayStreamInput(byte[] buf, int offset, int length) {
