diff --git a/docs/changelog/137032.yaml b/docs/changelog/137032.yaml
new file mode 100644
index 0000000000000..b569efd5cc343
--- /dev/null
+++ b/docs/changelog/137032.yaml
@@ -0,0 +1,7 @@
+pr: 137032
+summary: Fix handling equality with MV constants properly
+area: ES|QL
+type: bug
+issues:
+ - 136998
+ - 136939
diff --git a/x-pack/plugin/esql/qa/testFixtures/src/main/resources/folding.csv-spec b/x-pack/plugin/esql/qa/testFixtures/src/main/resources/folding.csv-spec
index 4e35e0ba9efcf..dd5e0e7fa3345 100644
--- a/x-pack/plugin/esql/qa/testFixtures/src/main/resources/folding.csv-spec
+++ b/x-pack/plugin/esql/qa/testFixtures/src/main/resources/folding.csv-spec
@@ -103,3 +103,69 @@ a:s    | b:s      | x:i | y:s          | z:s
 "some" | "string" | 4   | "somestring" | "anotherstring"
 ;
 
+mvBoolean_Equals_MultiValueConstant
+required_capability: fix_mv_constant_equals_field
+
+FROM employees
+| WHERE is_rehired == ([true, false])
+| KEEP is_rehired
+;
+warning:Line 2:9: evaluation of [is_rehired == ([true, false])] failed, treating result as null. Only first 20 failures recorded.
+warning:Line 2:9: java.lang.IllegalArgumentException: single-value function encountered multi-value
+
+    is_rehired:boolean
+;
+
+computedBoolean_Equals_MultiValueConstant
+required_capability: fix_mv_constant_equals_field
+
+FROM employees
+| EVAL x = gender IS NULL
+| WHERE x == [true, false]
+| KEEP x
+;
+warning:Line 3:9: evaluation of [x == [true, false]] failed, treating result as null. Only first 20 failures recorded.
+warning:Line 3:9: java.lang.IllegalArgumentException: single-value function encountered multi-value
+
+    x:boolean
+;
+
+computedBoolean_Equals_SingleMultiValueConstant
+required_capability: fix_mv_constant_equals_field
+
+FROM employees
+| EVAL x = gender IS NULL
+| WHERE x == [true]
+| KEEP x
+| LIMIT 1
+;
+
+    x:boolean
+    true
+;
+
+mvDouble_IN_MultiValueConstant
+required_capability: fix_mv_constant_equals_field
+
+FROM employees
+| WHERE salary_change IN ([1,2,3])
+| KEEP salary_change
+;
+warning:Line 2:9: evaluation of [salary_change IN ([1,2,3])] failed, treating result as null. Only first 20 failures recorded.
+warning:Line 2:9: java.lang.IllegalArgumentException: single-value function encountered multi-value
+
+    salary_change:double
+;
+
+mvDouble_Equals_MultiValueConstant
+required_capability: fix_mv_constant_equals_field
+
+FROM employees
+| WHERE salary_change == [1,2,3]
+| KEEP salary_change
+;
+warning:Line 2:9: evaluation of [salary_change == [1,2,3]] failed, treating result as null. Only first 20 failures recorded.
+warning:Line 2:9: java.lang.IllegalArgumentException: single-value function encountered multi-value
+
+    salary_change:double
+;
diff --git a/x-pack/plugin/esql/qa/testFixtures/src/main/resources/string.csv-spec b/x-pack/plugin/esql/qa/testFixtures/src/main/resources/string.csv-spec
index 4d9ddb83ae301..c1ec0bcd1f639 100644
--- a/x-pack/plugin/esql/qa/testFixtures/src/main/resources/string.csv-spec
+++ b/x-pack/plugin/esql/qa/testFixtures/src/main/resources/string.csv-spec
@@ -2582,6 +2582,54 @@ warning:Line 2:9: java.lang.IllegalArgumentException: single-value function enco
 2023-10-23T13:55:01.544Z|Connected to 10.1.0.1
 ;
 
+mvStringEqualsMultiValueConstant
+required_capability: fix_mv_constant_equals_field
+FROM mv_text
+| WHERE message == ["Connected to 10.1.0.1", "Banana"]
+| KEEP @timestamp, message
+;
+warning:Line 2:9: evaluation of [message == [\"Connected to 10.1.0.1\", \"Banana\"]] failed, treating result as null. Only first 20 failures recorded.
+warning:Line 2:9: java.lang.IllegalArgumentException: single-value function encountered multi-value
+
+    @timestamp:date     | message:text
+;
+
+mvString_IN_MultiValueConstant
+required_capability: fix_mv_constant_equals_field
+FROM mv_text
+| WHERE message IN (["Connected to 10.1.0.1", "Banana"])
+| KEEP @timestamp, message
+;
+warning:Line 2:9: evaluation of [message IN ([\"Connected to 10.1.0.1\", \"Banana\"])] failed, treating result as null. Only first 20 failures recorded.
+warning:Line 2:9: java.lang.IllegalArgumentException: single-value function encountered multi-value
+
+    @timestamp:date     | message:text
+;
+
+mvKeyword_IN_MultiValueConstant
+required_capability: fix_mv_constant_equals_field
+FROM employees
+| WHERE job_positions IN (["Tech Lead" , "Data Scientist", "Senior Team Lead"])
+| KEEP emp_no, job_positions
+;
+warning:Line 2:9: evaluation of [job_positions IN ([\"Tech Lead\" , \"Data Scientist\", \"Senior Team Lead\"])] failed, treating result as null. Only first 20 failures recorded.
+warning:Line 2:9: java.lang.IllegalArgumentException: single-value function encountered multi-value
+
+    emp_no:integer     | job_positions:keyword
+;
+
+mvKeywordEqualsMultiValueConstant
+required_capability: fix_mv_constant_equals_field
+FROM employees
+| WHERE job_positions == ["Tech Lead" , "Data Scientist", "Senior Team Lead"]
+| KEEP emp_no, job_positions
+;
+warning:Line 2:9: evaluation of [job_positions == [\"Tech Lead\" , \"Data Scientist\", \"Senior Team Lead\"]] failed, treating result as null. Only first 20 failures recorded.
+warning:Line 2:9: java.lang.IllegalArgumentException: single-value function encountered multi-value
+
+    emp_no:integer     | job_positions:keyword
+;
+
 url_encode sample for docs
 required_capability: url_encode
 
diff --git a/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlCapabilities.java b/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlCapabilities.java
index 56a5d350a20be..f457bd9c6ce1b 100644
--- a/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlCapabilities.java
+++ b/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlCapabilities.java
@@ -1580,6 +1580,7 @@ public enum Cap {
          */
         INLINE_STATS_WITH_NO_COLUMNS(INLINE_STATS.enabled),
 
+        FIX_MV_CONSTANT_EQUALS_FIELD,
         // Last capability should still have a comma for fewer merge conflicts when adding new ones :)
         // This comment prevents the semicolon from being on the previous capability when Spotless formats the file.
         ;
diff --git a/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/predicate/operator/comparison/Equals.java b/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/predicate/operator/comparison/Equals.java
index 63cda52368eb8..0ab0e3c20a2f9 100644
--- a/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/predicate/operator/comparison/Equals.java
+++ b/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/predicate/operator/comparison/Equals.java
@@ -27,6 +27,7 @@
 import org.elasticsearch.xpack.esql.querydsl.query.SingleValueQuery;
 
 import java.time.ZoneId;
+import java.util.Collection;
 import java.util.Map;
 
 public class Equals extends EsqlBinaryComparison implements Negatable<EsqlBinaryComparison> {
@@ -139,6 +140,12 @@ public Equals(Source source, Expression left, Expression right, ZoneId zoneId) {
     @Override
     public Translatable translatable(LucenePushdownPredicates pushdownPredicates) {
         if (right() instanceof Literal lit) {
+            // Multi-valued literals are not supported going further. This also makes sure that we are handling multi-valued literals with
+            // a "warning" header, as well (see EqualsKeywordsEvaluator, for example, where lhs and rhs are both dealt with equally when
+            // it comes to multi-value handling).
+            if (lit.value() instanceof Collection<?>) {
+                return Translatable.NO;
+            }
             if (left().dataType() == DataType.TEXT && left() instanceof FieldAttribute fa) {
                 if (pushdownPredicates.canUseEqualityOnSyntheticSourceDelegate(fa, ((BytesRef) lit.value()).utf8ToString())) {
                     return Translatable.YES_BUT_RECHECK_NEGATED;
