diff --git a/muted-tests.yml b/muted-tests.yml
index 2e2bdb22d6a70..13f6f7ac21a12 100644
--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -354,9 +354,6 @@ tests:
 - class: org.elasticsearch.search.CCSDuelIT
   method: testTermsAggsWithProfile
   issue: https://github.com/elastic/elasticsearch/issues/132880
-- class: org.elasticsearch.xpack.search.AsyncSearchErrorTraceIT
-  method: testAsyncSearchFailingQueryErrorTraceDefault
-  issue: https://github.com/elastic/elasticsearch/issues/133010
 - class: org.elasticsearch.xpack.security.authc.AuthenticationServiceTests
   method: testInvalidToken
   issue: https://github.com/elastic/elasticsearch/issues/133328
diff --git a/x-pack/plugin/async-search/src/internalClusterTest/java/org/elasticsearch/xpack/search/AsyncSearchErrorTraceIT.java b/x-pack/plugin/async-search/src/internalClusterTest/java/org/elasticsearch/xpack/search/AsyncSearchErrorTraceIT.java
index c132d38dd2792..83e7cf9e12096 100644
--- a/x-pack/plugin/async-search/src/internalClusterTest/java/org/elasticsearch/xpack/search/AsyncSearchErrorTraceIT.java
+++ b/x-pack/plugin/async-search/src/internalClusterTest/java/org/elasticsearch/xpack/search/AsyncSearchErrorTraceIT.java
@@ -76,7 +76,7 @@ public void testAsyncSearchFailingQueryErrorTraceDefault() throws Exception {
         createAsyncRequest.addParameter("keep_on_completion", "true");
         createAsyncRequest.addParameter("wait_for_completion_timeout", "0ms");
         Map<String, Object> createAsyncResponseEntity = performRequestAndGetResponseEntity(createAsyncRequest);
-        if (createAsyncResponseEntity.get("is_running").equals("true")) {
+        if (Boolean.TRUE.equals(createAsyncResponseEntity.get("is_running"))) {
             String asyncExecutionId = (String) createAsyncResponseEntity.get("id");
             Request getAsyncRequest = new Request("GET", "/_async_search/" + asyncExecutionId);
             awaitAsyncRequestDoneRunning(getAsyncRequest);
@@ -103,7 +103,7 @@ public void testAsyncSearchFailingQueryErrorTraceTrue() throws Exception {
         createAsyncRequest.addParameter("keep_on_completion", "true");
         createAsyncRequest.addParameter("wait_for_completion_timeout", "0ms");
         Map<String, Object> createAsyncResponseEntity = performRequestAndGetResponseEntity(createAsyncRequest);
-        if (createAsyncResponseEntity.get("is_running").equals("true")) {
+        if (Boolean.TRUE.equals(createAsyncResponseEntity.get("is_running"))) {
             String asyncExecutionId = (String) createAsyncResponseEntity.get("id");
             Request getAsyncRequest = new Request("GET", "/_async_search/" + asyncExecutionId);
             getAsyncRequest.addParameter("error_trace", "true");
@@ -131,7 +131,7 @@ public void testAsyncSearchFailingQueryErrorTraceFalse() throws Exception {
         createAsyncRequest.addParameter("keep_on_completion", "true");
         createAsyncRequest.addParameter("wait_for_completion_timeout", "0ms");
         Map<String, Object> createAsyncResponseEntity = performRequestAndGetResponseEntity(createAsyncRequest);
-        if (createAsyncResponseEntity.get("is_running").equals("true")) {
+        if (Boolean.TRUE.equals(createAsyncResponseEntity.get("is_running"))) {
             String asyncExecutionId = (String) createAsyncResponseEntity.get("id");
             Request getAsyncRequest = new Request("GET", "/_async_search/" + asyncExecutionId);
             getAsyncRequest.addParameter("error_trace", "false");
@@ -172,7 +172,7 @@ public void testDataNodeLogsStackTrace() throws Exception {
         try (var mockLog = MockLog.capture(SearchService.class)) {
             ErrorTraceHelper.addSeenLoggingExpectations(numShards, mockLog, errorTriggeringIndex);
             Map<String, Object> createAsyncResponseEntity = performRequestAndGetResponseEntity(createAsyncRequest);
-            if (createAsyncResponseEntity.get("is_running").equals("true")) {
+            if (Boolean.TRUE.equals(createAsyncResponseEntity.get("is_running"))) {
                 String asyncExecutionId = (String) createAsyncResponseEntity.get("id");
                 Request getAsyncRequest = new Request("GET", "/_async_search/" + asyncExecutionId);
                 // Use the same value of error_trace as the search request
@@ -206,7 +206,7 @@ public void testAsyncSearchFailingQueryErrorTraceFalseOnSubmitAndTrueOnGet() thr
         createAsyncSearchRequest.addParameter("keep_on_completion", "true");
         createAsyncSearchRequest.addParameter("wait_for_completion_timeout", "0ms");
         Map<String, Object> createAsyncResponseEntity = performRequestAndGetResponseEntity(createAsyncSearchRequest);
-        if (createAsyncResponseEntity.get("is_running").equals("true")) {
+        if (Boolean.TRUE.equals(createAsyncResponseEntity.get("is_running"))) {
             String asyncExecutionId = (String) createAsyncResponseEntity.get("id");
             Request getAsyncRequest = new Request("GET", "/_async_search/" + asyncExecutionId);
             getAsyncRequest.addParameter("error_trace", "true");
@@ -234,7 +234,7 @@ public void testAsyncSearchFailingQueryErrorTraceTrueOnSubmitAndFalseOnGet() thr
         createAsyncSearchRequest.addParameter("keep_on_completion", "true");
         createAsyncSearchRequest.addParameter("wait_for_completion_timeout", "0ms");
         Map<String, Object> createAsyncResponseEntity = performRequestAndGetResponseEntity(createAsyncSearchRequest);
-        if (createAsyncResponseEntity.get("is_running").equals("true")) {
+        if (Boolean.TRUE.equals(createAsyncResponseEntity.get("is_running"))) {
             String asyncExecutionId = (String) createAsyncResponseEntity.get("id");
             Request getAsyncRequest = new Request("GET", "/_async_search/" + asyncExecutionId);
             getAsyncRequest.addParameter("error_trace", "false");
