diff --git a/muted-tests.yml b/muted-tests.yml
index 667f99c046975..ece853d749214 100644
--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -372,9 +372,6 @@ tests:
 - class: org.elasticsearch.xpack.ml.integration.InferenceIT
   method: testInferClassificationModel
   issue: https://github.com/elastic/elasticsearch/issues/133448
-- class: org.elasticsearch.xpack.esql.action.CrossClusterQueryWithPartialResultsIT
-  method: testPartialResults
-  issue: https://github.com/elastic/elasticsearch/issues/131481
 - class: org.elasticsearch.xpack.ml.integration.ClassificationIT
   method: testWithCustomFeatureProcessors
   issue: https://github.com/elastic/elasticsearch/issues/134001
diff --git a/x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/action/CrossClusterQueryWithPartialResultsIT.java b/x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/action/CrossClusterQueryWithPartialResultsIT.java
index b1034b5130fe1..a202ede73be20 100644
--- a/x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/action/CrossClusterQueryWithPartialResultsIT.java
+++ b/x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/action/CrossClusterQueryWithPartialResultsIT.java
@@ -102,10 +102,10 @@ private void assertClusterSuccess(EsqlQueryResponse resp, String clusterAlias, i
 
     public void testPartialResults() throws Exception {
         populateIndices();
-        EsqlQueryRequest request = new EsqlQueryRequest();
-        request.query("FROM ok*,fail*,*:ok*,*:fail* | KEEP id, fail_me | LIMIT 1000");
-        request.includeCCSMetadata(randomBoolean());
         {
+            EsqlQueryRequest request = new EsqlQueryRequest();
+            request.query("FROM ok*,fail*,*:ok*,*:fail* | KEEP id, fail_me | LIMIT 1000");
+            request.includeCCSMetadata(randomBoolean());
             request.allowPartialResults(false);
             Exception error = expectThrows(Exception.class, () -> runQuery(request).close());
             error = EsqlTestUtils.unwrapIfWrappedInRemoteException(error);
@@ -113,6 +113,9 @@ public void testPartialResults() throws Exception {
             assertThat(error, instanceOf(IllegalStateException.class));
             assertThat(error.getMessage(), containsString("Accessing failing field"));
         }
+        EsqlQueryRequest request = new EsqlQueryRequest();
+        request.query("FROM ok*,fail*,*:ok*,*:fail* | KEEP id, fail_me | LIMIT 1000");
+        request.includeCCSMetadata(randomBoolean());
         request.allowPartialResults(true);
         try (var resp = runQuery(request)) {
             assertTrue(resp.isPartial());
@@ -134,6 +137,8 @@ public void testPartialResults() throws Exception {
             for (String cluster : List.of(LOCAL_CLUSTER, REMOTE_CLUSTER_1, REMOTE_CLUSTER_2)) {
                 assertClusterFailure(resp, cluster, "Accessing failing field");
             }
+        } catch (Exception e) {
+            throw new AssertionError("failed to execute query", e);
         }
     }
 
