diff --git a/spring-web/src/main/java/org/springframework/http/codec/CodecConfigurer.java b/spring-web/src/main/java/org/springframework/http/codec/CodecConfigurer.java
index 05d134171ec8..28c713a7bfff 100644
--- a/spring-web/src/main/java/org/springframework/http/codec/CodecConfigurer.java
+++ b/spring-web/src/main/java/org/springframework/http/codec/CodecConfigurer.java
@@ -206,7 +206,7 @@ default void jackson2SmileDecoder(Decoder<?> decoder) {
 		@Deprecated(since = "7.0", forRemoval = true)
 		default void jackson2SmileEncoder(Encoder<?> encoder) {
 			jacksonSmileEncoder(encoder);
-		};
+		}
 
 		/**
 		 * Override the default Protobuf {@code Decoder}.
diff --git a/spring-web/src/main/java/org/springframework/http/codec/support/BaseDefaultCodecs.java b/spring-web/src/main/java/org/springframework/http/codec/support/BaseDefaultCodecs.java
index d534c7154e10..8a4ab22d9b5f 100644
--- a/spring-web/src/main/java/org/springframework/http/codec/support/BaseDefaultCodecs.java
+++ b/spring-web/src/main/java/org/springframework/http/codec/support/BaseDefaultCodecs.java
@@ -827,7 +827,7 @@ else if (JACKSON_2_PRESENT) {
 
 	/**
 	 * Get or initialize a Jackson JSON decoder.
-	 * @deprecated in favor of {@link #getJacksonJsonEncoder()}
+	 * @deprecated in favor of {@link #getJacksonJsonDecoder()}
 	 */
 	@Deprecated(since = "7.0", forRemoval = true)
 	protected Decoder<?> getJackson2JsonDecoder() {
